{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e692c685",
   "metadata": {},
   "source": [
    "# package"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ac548cfc",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:02:54.481789Z",
     "start_time": "2022-08-11T11:02:50.759828Z"
    }
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import networkx as nx\n",
    "import copy\n",
    "import sympy\n",
    "from numpy.linalg import det\n",
    "import scipy\n",
    "from numpy.linalg import inv\n",
    "from tqdm import tqdm\n",
    "import random\n",
    "import scipy.stats\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "020bbe74",
   "metadata": {},
   "source": [
    "# network"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "4c9c76b1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:02:56.294998Z",
     "start_time": "2022-08-11T11:02:54.498720Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Name: \n",
      "Type: Graph\n",
      "Number of nodes: 15961\n",
      "Number of edges: 237535\n",
      "Average degree:  29.7644\n",
      "15961\n"
     ]
    }
   ],
   "source": [
    "network_file_dir = r'../../data/STRINGv11_07.txt'\n",
    "network_file = open(network_file_dir,mode='rb')\n",
    "network = nx.read_weighted_edgelist(network_file,create_using=nx.Graph())\n",
    "print(nx.info(network))\n",
    "node = network.nodes\n",
    "print(len(node))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "725dfbbc",
   "metadata": {},
   "source": [
    "# pathway"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6b48da0e",
   "metadata": {},
   "source": [
    "## KEGG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "db9876ea",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:02:57.480618Z",
     "start_time": "2022-08-11T11:02:56.311927Z"
    }
   },
   "outputs": [],
   "source": [
    "kegg_dir = 'D:/downloads/KEGG_2021_Human.xlsx'\n",
    "df_kegg = pd.read_excel(kegg_dir,header=None)\n",
    "df_kegg = df_kegg.fillna(0)\n",
    "df_kegg_list = df_kegg.values.tolist()\n",
    "for i in range(len(df_kegg_list)):\n",
    "    res = []\n",
    "    for con in df_kegg_list[i] :\n",
    "        if con == 0:\n",
    "            continue\n",
    "        if con not in res:\n",
    "            res.append(con)\n",
    "    df_kegg_list[i] = res\n",
    "    df_kegg_list[i].pop()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "ba5db1cd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:02:57.557411Z",
     "start_time": "2022-08-11T11:02:57.542421Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ABC transporters', 'ABCA2', 'ABCC4', 'ABCG8', 'ABCC5']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_kegg_list[0][:5]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5432d97d",
   "metadata": {},
   "source": [
    "# disease"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "74918c39",
   "metadata": {},
   "source": [
    "## COVID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d4a69ff0",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:25.191952Z",
     "start_time": "2022-08-11T11:03:25.174998Z"
    }
   },
   "outputs": [],
   "source": [
    "COVID_dir = '../../data/COVID-19_Related_Gene_Sets_2021.xlsx'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "e6d896b3",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:27.920504Z",
     "start_time": "2022-08-11T11:03:25.703215Z"
    }
   },
   "outputs": [],
   "source": [
    "COVID = pd.read_excel(COVID_dir,header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "dcaf95f7",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:28.014252Z",
     "start_time": "2022-08-11T11:03:27.985344Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>744</th>\n",
       "      <th>745</th>\n",
       "      <th>746</th>\n",
       "      <th>747</th>\n",
       "      <th>748</th>\n",
       "      <th>749</th>\n",
       "      <th>750</th>\n",
       "      <th>751</th>\n",
       "      <th>752</th>\n",
       "      <th>753</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>100 gene most correlated to LDLRAD4 based on A...</td>\n",
       "      <td>ARFGEF3</td>\n",
       "      <td>SLC12A6</td>\n",
       "      <td>IL10RA</td>\n",
       "      <td>RHOH</td>\n",
       "      <td>NFATC2</td>\n",
       "      <td>NFATC1</td>\n",
       "      <td>LNPEP</td>\n",
       "      <td>ARHGAP27</td>\n",
       "      <td>PLXDC2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>152 Down Genes from GEN3VA Mouse Lung; Accessi...</td>\n",
       "      <td>MYLIP</td>\n",
       "      <td>RPL22</td>\n",
       "      <td>RPL21</td>\n",
       "      <td>CNBP</td>\n",
       "      <td>RPL23</td>\n",
       "      <td>RPS7</td>\n",
       "      <td>DEK</td>\n",
       "      <td>FBXO16</td>\n",
       "      <td>ATP1B1</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>178 positive hits from CRISPR screen of Heaton...</td>\n",
       "      <td>GAMT</td>\n",
       "      <td>ZNF287</td>\n",
       "      <td>SZT2</td>\n",
       "      <td>SCARA3</td>\n",
       "      <td>BRF2</td>\n",
       "      <td>YIPF3</td>\n",
       "      <td>LRRN3</td>\n",
       "      <td>SCARA5</td>\n",
       "      <td>SLC12A2</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>184 proteins up-regualated at 0h in SARS-CoV-2...</td>\n",
       "      <td>MAP3K2</td>\n",
       "      <td>TRAPPC10</td>\n",
       "      <td>CPSF3</td>\n",
       "      <td>TJAP1</td>\n",
       "      <td>ARHGAP29</td>\n",
       "      <td>EVA1A</td>\n",
       "      <td>GAPVD1</td>\n",
       "      <td>SRRM1</td>\n",
       "      <td>SYTL4</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>192 negative hits from CRISPR screen of Heaton...</td>\n",
       "      <td>FBXW12</td>\n",
       "      <td>GINS4</td>\n",
       "      <td>USP4</td>\n",
       "      <td>CRYBB3</td>\n",
       "      <td>ASRGL1</td>\n",
       "      <td>THEMIS</td>\n",
       "      <td>WDR74</td>\n",
       "      <td>GOLGA8O</td>\n",
       "      <td>HIPK4</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>Top 500 upregulated genes in mouse lung with S...</td>\n",
       "      <td>CDK18</td>\n",
       "      <td>RFX1</td>\n",
       "      <td>SCRIB</td>\n",
       "      <td>GTPBP2</td>\n",
       "      <td>EML3</td>\n",
       "      <td>CAPN15</td>\n",
       "      <td>HNRNPL</td>\n",
       "      <td>HNRNPD</td>\n",
       "      <td>DUS3L</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>474</th>\n",
       "      <td>Top 500 upregulated genes in mouse spleen with...</td>\n",
       "      <td>SCRIB</td>\n",
       "      <td>TMEM80</td>\n",
       "      <td>GTPBP2</td>\n",
       "      <td>NISCH</td>\n",
       "      <td>CAPN15</td>\n",
       "      <td>TEP1</td>\n",
       "      <td>ARMC5</td>\n",
       "      <td>DUS3L</td>\n",
       "      <td>CDK16</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>475</th>\n",
       "      <td>Top 772 negative hits from genome-wide CRISPR ...</td>\n",
       "      <td>STARD3</td>\n",
       "      <td>ISY1-RAB43</td>\n",
       "      <td>MECR</td>\n",
       "      <td>LACRT</td>\n",
       "      <td>PRELID1</td>\n",
       "      <td>AP1S1</td>\n",
       "      <td>RPL24</td>\n",
       "      <td>WDR75</td>\n",
       "      <td>GTPBP6</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>476</th>\n",
       "      <td>Top 797 negative hits from genome-wide CRISPR ...</td>\n",
       "      <td>STARD3</td>\n",
       "      <td>LRRC37A3</td>\n",
       "      <td>PRELID1</td>\n",
       "      <td>BTBD18</td>\n",
       "      <td>RPL21</td>\n",
       "      <td>RPL23</td>\n",
       "      <td>LRRC37A</td>\n",
       "      <td>ZFYVE26</td>\n",
       "      <td>HSPE1</td>\n",
       "      <td>...</td>\n",
       "      <td>FLCN</td>\n",
       "      <td>RBM4</td>\n",
       "      <td>RMND5A</td>\n",
       "      <td>POLD1</td>\n",
       "      <td>POLD2</td>\n",
       "      <td>SPATA24</td>\n",
       "      <td>NOS2</td>\n",
       "      <td>CHCHD1</td>\n",
       "      <td>RNF113A</td>\n",
       "      <td>SMNDC1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>477</th>\n",
       "      <td>Up regulated gene from COVID-19 infected bronc...</td>\n",
       "      <td>CADM2</td>\n",
       "      <td>WDPCP</td>\n",
       "      <td>TEX35</td>\n",
       "      <td>PLEKHA5</td>\n",
       "      <td>SHISA9</td>\n",
       "      <td>AIG1</td>\n",
       "      <td>LPAR6</td>\n",
       "      <td>LACC1</td>\n",
       "      <td>LOC112268387</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>478 rows × 754 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   0        1           2    \\\n",
       "0    100 gene most correlated to LDLRAD4 based on A...  ARFGEF3     SLC12A6   \n",
       "1    152 Down Genes from GEN3VA Mouse Lung; Accessi...    MYLIP       RPL22   \n",
       "2    178 positive hits from CRISPR screen of Heaton...     GAMT      ZNF287   \n",
       "3    184 proteins up-regualated at 0h in SARS-CoV-2...   MAP3K2    TRAPPC10   \n",
       "4    192 negative hits from CRISPR screen of Heaton...   FBXW12       GINS4   \n",
       "..                                                 ...      ...         ...   \n",
       "473  Top 500 upregulated genes in mouse lung with S...    CDK18        RFX1   \n",
       "474  Top 500 upregulated genes in mouse spleen with...    SCRIB      TMEM80   \n",
       "475  Top 772 negative hits from genome-wide CRISPR ...   STARD3  ISY1-RAB43   \n",
       "476  Top 797 negative hits from genome-wide CRISPR ...   STARD3    LRRC37A3   \n",
       "477  Up regulated gene from COVID-19 infected bronc...    CADM2       WDPCP   \n",
       "\n",
       "         3        4         5       6        7         8             9    ...  \\\n",
       "0     IL10RA     RHOH    NFATC2  NFATC1    LNPEP  ARHGAP27        PLXDC2  ...   \n",
       "1      RPL21     CNBP     RPL23    RPS7      DEK    FBXO16        ATP1B1  ...   \n",
       "2       SZT2   SCARA3      BRF2   YIPF3    LRRN3    SCARA5       SLC12A2  ...   \n",
       "3      CPSF3    TJAP1  ARHGAP29   EVA1A   GAPVD1     SRRM1         SYTL4  ...   \n",
       "4       USP4   CRYBB3    ASRGL1  THEMIS    WDR74   GOLGA8O         HIPK4  ...   \n",
       "..       ...      ...       ...     ...      ...       ...           ...  ...   \n",
       "473    SCRIB   GTPBP2      EML3  CAPN15   HNRNPL    HNRNPD         DUS3L  ...   \n",
       "474   GTPBP2    NISCH    CAPN15    TEP1    ARMC5     DUS3L         CDK16  ...   \n",
       "475     MECR    LACRT   PRELID1   AP1S1    RPL24     WDR75        GTPBP6  ...   \n",
       "476  PRELID1   BTBD18     RPL21   RPL23  LRRC37A   ZFYVE26         HSPE1  ...   \n",
       "477    TEX35  PLEKHA5    SHISA9    AIG1    LPAR6     LACC1  LOC112268387  ...   \n",
       "\n",
       "      744   745     746    747    748      749   750     751      752     753  \n",
       "0     NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "1     NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "2     NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "3     NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "4     NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "..    ...   ...     ...    ...    ...      ...   ...     ...      ...     ...  \n",
       "473   NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "474   NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "475   NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "476  FLCN  RBM4  RMND5A  POLD1  POLD2  SPATA24  NOS2  CHCHD1  RNF113A  SMNDC1  \n",
       "477   NaN   NaN     NaN    NaN    NaN      NaN   NaN     NaN      NaN     NaN  \n",
       "\n",
       "[478 rows x 754 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COVID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "f7e0db48",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:31.139030Z",
     "start_time": "2022-08-11T11:03:29.667964Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "478 478\n"
     ]
    }
   ],
   "source": [
    "COVID = COVID.fillna(0)\n",
    "COVID_list = COVID.values.tolist()\n",
    "for i in range(len(COVID_list)):\n",
    "    res = []\n",
    "    for con in COVID_list[i] :\n",
    "        if con not in res:\n",
    "            res.append(con)\n",
    "    COVID_list[i] = res\n",
    "    COVID_list[i].pop()\n",
    "print(len(COVID),len(COVID_list))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "7da9699e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:31.308575Z",
     "start_time": "2022-08-11T11:03:31.293586Z"
    }
   },
   "outputs": [],
   "source": [
    "COVID_merge = []\n",
    "COVID_merge.extend(COVID_list[6])\n",
    "COVID_merge.extend(COVID_list[9])\n",
    "COVID_merge.extend(COVID_list[11])\n",
    "COVID_merge.extend(COVID_list[13])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "103f1e70",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:31.370929Z",
     "start_time": "2022-08-11T11:03:31.356418Z"
    },
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#print(len(COVID_list[36]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8cb98ff1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:31.465709Z",
     "start_time": "2022-08-11T11:03:31.459721Z"
    },
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "28\n",
      "50\n",
      "104\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'21 Positive hits from CRISPR screen from Hoffmann et al. (PMID 32935098) in Huh-7.5 cells at 37 deg C',\n",
       " '26 Negative hits from CRISPR screen from Hoffmann et al. (PMID 32935098) in Huh-7.5 cells at 33 deg C',\n",
       " '27 Positive hits from CRISPR screen from Hoffmann et al. (PMID 32935098) in Huh-7.5 cells at 33 deg C',\n",
       " '28 Negative hits from CRISPR screen from Hoffmann et al. (PMID 32935098) in Huh-7.5 cells at 37 deg C',\n",
       " 'ACAD9',\n",
       " 'ALG5',\n",
       " 'ALG8',\n",
       " 'ANAPC5',\n",
       " 'AP3B1',\n",
       " 'BCS1L',\n",
       " 'BRD2',\n",
       " 'CHMP2A',\n",
       " 'COMT',\n",
       " 'CSDE1',\n",
       " 'CSNK2B',\n",
       " 'CUL2',\n",
       " 'CWC27',\n",
       " 'DCAF7',\n",
       " 'DDX10',\n",
       " 'DNAJC11',\n",
       " 'DNAJC19',\n",
       " 'DPH5',\n",
       " 'ECSIT',\n",
       " 'EIF4E2',\n",
       " 'ELOC',\n",
       " 'EMC1',\n",
       " 'FAM98A',\n",
       " 'FBL',\n",
       " 'GIGYF2',\n",
       " 'GPAA1',\n",
       " 'GRPEL1',\n",
       " 'HECTD1',\n",
       " 'HS2ST1',\n",
       " 'INTS4',\n",
       " 'LARP7',\n",
       " 'MEPCE',\n",
       " 'MIB1',\n",
       " 'MOGS',\n",
       " 'MRPS2',\n",
       " 'MRPS25',\n",
       " 'MRPS27',\n",
       " 'MRPS5',\n",
       " 'NARS2',\n",
       " 'NDUFAF1',\n",
       " 'NDUFAF2',\n",
       " 'NDUFB9',\n",
       " 'NOL10',\n",
       " 'NPC2',\n",
       " 'NUTF2',\n",
       " 'PABPC1',\n",
       " 'PCF11',\n",
       " 'PIGO',\n",
       " 'PITRM1',\n",
       " 'PLEKHF2',\n",
       " 'PRIM1',\n",
       " 'PRIM2',\n",
       " 'PSMD4',\n",
       " 'RAB10',\n",
       " 'RAB14',\n",
       " 'RAB2A',\n",
       " 'RAB7A',\n",
       " 'RBX1',\n",
       " 'RHOA',\n",
       " 'SCAP',\n",
       " 'SEPSECS',\n",
       " 'SLC30A9',\n",
       " 'SLC35B1',\n",
       " 'SRP19',\n",
       " 'SRP54',\n",
       " 'SRP72',\n",
       " 'TARS2',\n",
       " 'TBCA',\n",
       " 'TOMM70',\n",
       " 'TPT1',\n",
       " 'UBAP2L',\n",
       " 'VPS11',\n",
       " 'VPS39',\n",
       " 'WASHC4',\n",
       " 'ZNF131'}"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(COVID_list[290]))\n",
    "print(len(COVID_list[36]))\n",
    "print(len(COVID_merge))\n",
    "ace2 = set(COVID_list[290])\n",
    "crispr = set(COVID_list[36])\n",
    "function = set(COVID_merge)\n",
    "function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "c84bce34",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:31.995535Z",
     "start_time": "2022-08-11T11:03:31.991575Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'CTSL', 'ACE2'}\n"
     ]
    }
   ],
   "source": [
    "print(ace2 & crispr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f1ae7917",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:32.651131Z",
     "start_time": "2022-08-11T11:03:32.646108Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'RAB7A', 'NPC2'}\n"
     ]
    }
   ],
   "source": [
    "print(ace2 & function)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "c792dabd",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:33.107929Z",
     "start_time": "2022-08-11T11:03:33.094965Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "set()\n"
     ]
    }
   ],
   "source": [
    "print(crispr & function)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a8913a9e",
   "metadata": {},
   "source": [
    "## CGC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b32e7c98",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:35.289416Z",
     "start_time": "2022-08-11T11:03:34.995623Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene Symbol</th>\n",
       "      <th>Name</th>\n",
       "      <th>Entrez GeneId</th>\n",
       "      <th>Genome Location</th>\n",
       "      <th>Tier</th>\n",
       "      <th>Hallmark</th>\n",
       "      <th>Chr Band</th>\n",
       "      <th>Somatic</th>\n",
       "      <th>Germline</th>\n",
       "      <th>Tumour Types(Somatic)</th>\n",
       "      <th>Tumour Types(Germline)</th>\n",
       "      <th>Cancer Syndrome</th>\n",
       "      <th>Tissue Type</th>\n",
       "      <th>Molecular Genetics</th>\n",
       "      <th>Role in Cancer</th>\n",
       "      <th>Mutation Types</th>\n",
       "      <th>Translocation Partner</th>\n",
       "      <th>Other Germline Mut</th>\n",
       "      <th>Other Syndrome</th>\n",
       "      <th>Synonyms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A1CF</td>\n",
       "      <td>APOBEC1 complementation factor</td>\n",
       "      <td>29974.0</td>\n",
       "      <td>10:50799421-50885675</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.23</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>melanoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>oncogene</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>29974,A1CF,ACF,ACF64,ACF65,APOBEC1CF,ASP,ENSG0...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABI1</td>\n",
       "      <td>abl-interactor 1</td>\n",
       "      <td>10006.0</td>\n",
       "      <td>10:26746593-26860935</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>12.10</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>TSG, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>KMT2A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10006,ABI-1,ABI1,E3B1,ENSG00000136754.17,Q8IZP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>v-abl Abelson murine leukemia viral oncogene h...</td>\n",
       "      <td>25.0</td>\n",
       "      <td>9:130713946-130885683</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>34.12</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CML, ALL, T-ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T, Mis</td>\n",
       "      <td>BCR, ETV6, NUP214</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25,ABL,ABL1,ENSG00000097007.17,JTK7,P00519,c-A...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL2</td>\n",
       "      <td>c-abl oncogene 2, non-receptor tyrosine kinase</td>\n",
       "      <td>27.0</td>\n",
       "      <td>1:179099327-179229601</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25.20</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AML</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>ETV6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>27,ABL2,ABLL,ARG,ENSG00000143322.19,P42684</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ACKR3</td>\n",
       "      <td>atypical chemokine receptor 3</td>\n",
       "      <td>57007.0</td>\n",
       "      <td>2:236569641-236582358</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>37.30</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lipoma</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>M</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>HMGA2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>57007,ACKR3,CMKOR1,CXCR7,ENSG00000144476.5,GPR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>718</th>\n",
       "      <td>ZNF429</td>\n",
       "      <td>zinc finger protein 429</td>\n",
       "      <td>353088.0</td>\n",
       "      <td>19:21505564-21538078</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.00</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GBM</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>O</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>353088,ENSG00000197013.9,Q86V71,ZNF429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>719</th>\n",
       "      <td>ZNF479</td>\n",
       "      <td>zinc finger protein 479</td>\n",
       "      <td>90827.0</td>\n",
       "      <td>7:57119614-57139864</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.20</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>lung cancer, bladder carcinoma, prostate carci...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>90827,ENSG00000185177.12,KR19,Q96JC4,ZNF479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>720</th>\n",
       "      <td>ZNF521</td>\n",
       "      <td>zinc finger protein 521</td>\n",
       "      <td>25925.0</td>\n",
       "      <td>18:25061926-25352152</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11.20</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Dom</td>\n",
       "      <td>oncogene, fusion</td>\n",
       "      <td>T</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25925,EHZF,ENSG00000198795.10,Evi3,Q96K83,ZNF521</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>721</th>\n",
       "      <td>ZNRF3</td>\n",
       "      <td>zinc and ring finger 3</td>\n",
       "      <td>84133.0</td>\n",
       "      <td>22:28883592-29057487</td>\n",
       "      <td>2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.10</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>colorectal cancer, adrenocortical carcinoma, g...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>E</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TSG</td>\n",
       "      <td>N, F, Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>84133,BK747E2.3,ENSG00000183579.15,FLJ22057,KI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>722</th>\n",
       "      <td>ZRSR2</td>\n",
       "      <td>zinc finger (CCCH type), RNA-binding motif and...</td>\n",
       "      <td>8233.0</td>\n",
       "      <td>X:15790472-15823260</td>\n",
       "      <td>1</td>\n",
       "      <td>Yes</td>\n",
       "      <td>22.20</td>\n",
       "      <td>yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MDS, CLL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>L</td>\n",
       "      <td>Rec</td>\n",
       "      <td>TSG</td>\n",
       "      <td>F, S, Mis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8233,ENSG00000169249.12,Q15696,U2AF1-RS2,U2AF1...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>723 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene Symbol                                               Name  \\\n",
       "0          A1CF                     APOBEC1 complementation factor   \n",
       "1          ABI1                                   abl-interactor 1   \n",
       "2          ABL1  v-abl Abelson murine leukemia viral oncogene h...   \n",
       "3          ABL2     c-abl oncogene 2, non-receptor tyrosine kinase   \n",
       "4         ACKR3                      atypical chemokine receptor 3   \n",
       "..          ...                                                ...   \n",
       "718      ZNF429                            zinc finger protein 429   \n",
       "719      ZNF479                            zinc finger protein 479   \n",
       "720      ZNF521                            zinc finger protein 521   \n",
       "721       ZNRF3                             zinc and ring finger 3   \n",
       "722       ZRSR2  zinc finger (CCCH type), RNA-binding motif and...   \n",
       "\n",
       "     Entrez GeneId        Genome Location  Tier Hallmark  Chr Band Somatic  \\\n",
       "0          29974.0   10:50799421-50885675     2      NaN     11.23     yes   \n",
       "1          10006.0   10:26746593-26860935     1      Yes     12.10     yes   \n",
       "2             25.0  9:130713946-130885683     1      Yes     34.12     yes   \n",
       "3             27.0  1:179099327-179229601     1      NaN     25.20     yes   \n",
       "4          57007.0  2:236569641-236582358     1      Yes     37.30     yes   \n",
       "..             ...                    ...   ...      ...       ...     ...   \n",
       "718       353088.0   19:21505564-21538078     2      NaN     12.00     yes   \n",
       "719        90827.0    7:57119614-57139864     2      NaN     11.20     yes   \n",
       "720        25925.0   18:25061926-25352152     1      NaN     11.20     yes   \n",
       "721        84133.0   22:28883592-29057487     2      NaN     12.10     yes   \n",
       "722         8233.0    X:15790472-15823260     1      Yes     22.20     yes   \n",
       "\n",
       "    Germline                              Tumour Types(Somatic)  \\\n",
       "0        NaN                                           melanoma   \n",
       "1        NaN                                                AML   \n",
       "2        NaN                                    CML, ALL, T-ALL   \n",
       "3        NaN                                                AML   \n",
       "4        NaN                                             lipoma   \n",
       "..       ...                                                ...   \n",
       "718      NaN                                                GBM   \n",
       "719      NaN  lung cancer, bladder carcinoma, prostate carci...   \n",
       "720      NaN                                                ALL   \n",
       "721      NaN  colorectal cancer, adrenocortical carcinoma, g...   \n",
       "722      NaN                                           MDS, CLL   \n",
       "\n",
       "    Tumour Types(Germline) Cancer Syndrome Tissue Type Molecular Genetics  \\\n",
       "0                      NaN             NaN           E                NaN   \n",
       "1                      NaN             NaN           L                Dom   \n",
       "2                      NaN             NaN           L                Dom   \n",
       "3                      NaN             NaN           L                Dom   \n",
       "4                      NaN             NaN           M                Dom   \n",
       "..                     ...             ...         ...                ...   \n",
       "718                    NaN             NaN           O                NaN   \n",
       "719                    NaN             NaN           E                NaN   \n",
       "720                    NaN             NaN           L                Dom   \n",
       "721                    NaN             NaN           E                NaN   \n",
       "722                    NaN             NaN           L                Rec   \n",
       "\n",
       "       Role in Cancer Mutation Types Translocation Partner Other Germline Mut  \\\n",
       "0            oncogene            Mis                   NaN                NaN   \n",
       "1         TSG, fusion              T                 KMT2A                NaN   \n",
       "2    oncogene, fusion         T, Mis     BCR, ETV6, NUP214                NaN   \n",
       "3    oncogene, fusion              T                  ETV6                NaN   \n",
       "4    oncogene, fusion              T                 HMGA2                NaN   \n",
       "..                ...            ...                   ...                ...   \n",
       "718               NaN            Mis                   NaN                NaN   \n",
       "719               NaN            Mis                   NaN                NaN   \n",
       "720  oncogene, fusion              T                  PAX5                NaN   \n",
       "721               TSG      N, F, Mis                   NaN                NaN   \n",
       "722               TSG      F, S, Mis                   NaN                NaN   \n",
       "\n",
       "    Other Syndrome                                           Synonyms  \n",
       "0              NaN  29974,A1CF,ACF,ACF64,ACF65,APOBEC1CF,ASP,ENSG0...  \n",
       "1              NaN  10006,ABI-1,ABI1,E3B1,ENSG00000136754.17,Q8IZP...  \n",
       "2              NaN  25,ABL,ABL1,ENSG00000097007.17,JTK7,P00519,c-A...  \n",
       "3              NaN         27,ABL2,ABLL,ARG,ENSG00000143322.19,P42684  \n",
       "4              NaN  57007,ACKR3,CMKOR1,CXCR7,ENSG00000144476.5,GPR...  \n",
       "..             ...                                                ...  \n",
       "718            NaN             353088,ENSG00000197013.9,Q86V71,ZNF429  \n",
       "719            NaN        90827,ENSG00000185177.12,KR19,Q96JC4,ZNF479  \n",
       "720            NaN   25925,EHZF,ENSG00000198795.10,Evi3,Q96K83,ZNF521  \n",
       "721            NaN  84133,BK747E2.3,ENSG00000183579.15,FLJ22057,KI...  \n",
       "722            NaN  8233,ENSG00000169249.12,Q15696,U2AF1-RS2,U2AF1...  \n",
       "\n",
       "[723 rows x 20 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cancer_gene_census = pd.read_csv('D:/downloads/Census_allMon Aug 30 16_37_41 2021.csv')\n",
    "cancer_gene_census"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc4b5074",
   "metadata": {},
   "source": [
    "## NCG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ac9bb5bd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "2107a7f8",
   "metadata": {},
   "source": [
    "## driverdbv3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "a4de6c08",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:31:04.765027Z",
     "start_time": "2022-08-11T11:31:04.733114Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cancer_project</th>\n",
       "      <th>cancer_type_abbr</th>\n",
       "      <th>tool</th>\n",
       "      <th>driver_gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Adrenocortical_carcinoma(TCGA,US)</td>\n",
       "      <td>ACC</td>\n",
       "      <td>activedriver</td>\n",
       "      <td>APC, USF3, TMCC1, CXXC1, CPS1, KIAA1671, ADNP,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Adrenocortical_carcinoma(TCGA,US)</td>\n",
       "      <td>ACC</td>\n",
       "      <td>comet</td>\n",
       "      <td>ANO2, ANK3, STAB1, CLTB, HUWE1, EPG5, CHD2, RB...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Adrenocortical_carcinoma(TCGA,US)</td>\n",
       "      <td>ACC</td>\n",
       "      <td>dendrix</td>\n",
       "      <td>NLRP12, GOLGA4, NAV3, UNC13C, CMYA5, CCDC168, ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Adrenocortical_carcinoma(TCGA,US)</td>\n",
       "      <td>ACC</td>\n",
       "      <td>driverml</td>\n",
       "      <td>EXPH5, EML6, TP53, MUC4, NUP188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Adrenocortical_carcinoma(TCGA,US)</td>\n",
       "      <td>ACC</td>\n",
       "      <td>e-driver</td>\n",
       "      <td>XIRP2, SVEP1, CSMD3, CCDC168, HMCN1, VWF, GRID...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>367</th>\n",
       "      <td>Uveal_melanoma(TCGA,US)</td>\n",
       "      <td>UVM</td>\n",
       "      <td>msea</td>\n",
       "      <td>EIF1AX, SF3B1, GNAQ, GNA11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>368</th>\n",
       "      <td>Uveal_melanoma(TCGA,US)</td>\n",
       "      <td>UVM</td>\n",
       "      <td>mutex</td>\n",
       "      <td>GNAQ, GNA11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>369</th>\n",
       "      <td>Uveal_melanoma(TCGA,US)</td>\n",
       "      <td>UVM</td>\n",
       "      <td>mutsigcv</td>\n",
       "      <td>GNA11, HECW2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>370</th>\n",
       "      <td>Uveal_melanoma(TCGA,US)</td>\n",
       "      <td>UVM</td>\n",
       "      <td>netbox</td>\n",
       "      <td>BAP1, COL14A1, CYSLTR2, EIF1AX, GNA11, GNAQ, M...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>371</th>\n",
       "      <td>Uveal_melanoma(TCGA,US)</td>\n",
       "      <td>UVM</td>\n",
       "      <td>oncodriveclust</td>\n",
       "      <td>GNAQ, GNA11, SF3B1, EIF1AX</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>372 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        cancer_project cancer_type_abbr            tool  \\\n",
       "0    Adrenocortical_carcinoma(TCGA,US)              ACC    activedriver   \n",
       "1    Adrenocortical_carcinoma(TCGA,US)              ACC           comet   \n",
       "2    Adrenocortical_carcinoma(TCGA,US)              ACC         dendrix   \n",
       "3    Adrenocortical_carcinoma(TCGA,US)              ACC        driverml   \n",
       "4    Adrenocortical_carcinoma(TCGA,US)              ACC        e-driver   \n",
       "..                                 ...              ...             ...   \n",
       "367            Uveal_melanoma(TCGA,US)              UVM            msea   \n",
       "368            Uveal_melanoma(TCGA,US)              UVM           mutex   \n",
       "369            Uveal_melanoma(TCGA,US)              UVM        mutsigcv   \n",
       "370            Uveal_melanoma(TCGA,US)              UVM          netbox   \n",
       "371            Uveal_melanoma(TCGA,US)              UVM  oncodriveclust   \n",
       "\n",
       "                                           driver_gene  \n",
       "0    APC, USF3, TMCC1, CXXC1, CPS1, KIAA1671, ADNP,...  \n",
       "1    ANO2, ANK3, STAB1, CLTB, HUWE1, EPG5, CHD2, RB...  \n",
       "2    NLRP12, GOLGA4, NAV3, UNC13C, CMYA5, CCDC168, ...  \n",
       "3                      EXPH5, EML6, TP53, MUC4, NUP188  \n",
       "4    XIRP2, SVEP1, CSMD3, CCDC168, HMCN1, VWF, GRID...  \n",
       "..                                                 ...  \n",
       "367                         EIF1AX, SF3B1, GNAQ, GNA11  \n",
       "368                                        GNAQ, GNA11  \n",
       "369                                       GNA11, HECW2  \n",
       "370  BAP1, COL14A1, CYSLTR2, EIF1AX, GNA11, GNAQ, M...  \n",
       "371                         GNAQ, GNA11, SF3B1, EIF1AX  \n",
       "\n",
       "[372 rows x 4 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DriverDB_dir = r'../../data/mutation_download_tab.csv'\n",
    "driverdb = pd.read_csv(DriverDB_dir,delimiter='\\t')\n",
    "driverdb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "b3a4c2b4",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:32:08.722395Z",
     "start_time": "2022-08-11T11:32:08.714442Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'activedriver'"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "driverdb.tool[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "608f4dfc",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:37.694615Z",
     "start_time": "2022-08-11T11:03:37.606879Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<pandas.core.groupby.generic.DataFrameGroupBy object at 0x0000022D68963BB0>\n",
      "33\n",
      "33\n",
      "377\n",
      "10\n",
      "38\n",
      "16\n",
      "33\n"
     ]
    }
   ],
   "source": [
    "driverdb_group_cancertype = driverdb.groupby('cancer_type_abbr')\n",
    "print(driverdb_group_cancertype)\n",
    "\n",
    "GeneList_from_cancer_type_count_ToolPredictive = []\n",
    "number_from_cancer_type_count = []\n",
    "gene_set_from_cancer_type = []\n",
    "for i in driverdb_group_cancertype.first().index:\n",
    "    c_number = []\n",
    "    tmp_2 = []\n",
    "    cancer_type = []\n",
    "    tmp = ((driverdb_group_cancertype.get_group(i).driver_gene).str.split(', ')).tolist()\n",
    "    for content in tmp:\n",
    "        tmp_2.extend(content)\n",
    "        cancer_type.append(content)\n",
    "        c_number.append(len(content))\n",
    "    tmp_2 = set(tmp_2)\n",
    "    gene_set_from_cancer_type.append(cancer_type)\n",
    "    number_from_cancer_type_count.append(c_number)\n",
    "    GeneList_from_cancer_type_count_ToolPredictive.append(tmp_2)\n",
    "print(len(GeneList_from_cancer_type_count_ToolPredictive))\n",
    "print(len(number_from_cancer_type_count))\n",
    "print(len(GeneList_from_cancer_type_count_ToolPredictive[0]))\n",
    "print(len(gene_set_from_cancer_type[0]))\n",
    "print(len(gene_set_from_cancer_type[0][0]))\n",
    "print(len(gene_set_from_cancer_type[0][4]))\n",
    "print(len(gene_set_from_cancer_type))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "ed67ce3a",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:38.607911Z",
     "start_time": "2022-08-11T11:03:38.588961Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['APC',\n",
       " 'USF3',\n",
       " 'TMCC1',\n",
       " 'CXXC1',\n",
       " 'CPS1',\n",
       " 'KIAA1671',\n",
       " 'ADNP',\n",
       " 'COBL',\n",
       " 'C9',\n",
       " 'CTNNB1',\n",
       " 'FOXO1',\n",
       " 'HLTF',\n",
       " 'FADD',\n",
       " 'TRIP13',\n",
       " 'TRPC6',\n",
       " 'SMC4',\n",
       " 'MAP1B',\n",
       " 'RANBP2',\n",
       " 'ASCC3',\n",
       " 'GCN1',\n",
       " 'MDC1',\n",
       " 'CEP170',\n",
       " 'ANKS1A',\n",
       " 'MYH4',\n",
       " 'MYH2',\n",
       " 'CHD7',\n",
       " 'SMPD4',\n",
       " 'TAOK1',\n",
       " 'ARHGAP21',\n",
       " 'ANK3',\n",
       " 'SCAF1',\n",
       " 'PLA2G4A',\n",
       " 'JMJD1C',\n",
       " 'MYO1G',\n",
       " 'TNKS1BP1',\n",
       " 'FHAD1',\n",
       " 'XIRP2',\n",
       " 'FRMD7']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_set_from_cancer_type[0][0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "930255a9",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "# set2gaussian output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3383909e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:42.366520Z",
     "start_time": "2022-08-11T11:03:42.141357Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0                           Ras signaling pathway\n",
       "1      Synthesis and degradation of ketone bodies\n",
       "2                           Tryptophan metabolism\n",
       "3                             Cellular senescence\n",
       "4                                        Melanoma\n",
       "                          ...                    \n",
       "315                         Fatty acid elongation\n",
       "316                            Mineral absorption\n",
       "317                                 Legionellosis\n",
       "318            Vascular smooth muscle contraction\n",
       "319                            Nicotine addiction\n",
       "Name: 0, Length: 320, dtype: object"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kegg_gene_set_name = pd.read_table('D:/downloads/HighConfidence_kegg/kegg_300/gene_set_name_kegg.txt',header=None)\n",
    "kegg_gene_set_name = kegg_gene_set_name[0]\n",
    "kegg_gene_set_name\n",
    "#reactome_gene_set_name['name'] = reactome_gene_set_name[0]\n",
    "#del reactome_gene_set_name[0]\n",
    "#reactome_gene_set_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "ca578334",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:43.430942Z",
     "start_time": "2022-08-11T11:03:42.909970Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "320"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mu = np.load('D:/downloads/HighConfidence_kegg/kegg_300/output_embed__pathway_path_mu.out.npy')\n",
    "#cov = np.load('D:/downloads/HighConfidence_kegg/kegg_300/output_embed__pathway_path_cov.out.npy')\n",
    "g2g = np.load('D:/downloads/HighConfidence_kegg/kegg_300/output_embed__pathway_g2g_node_emb.out.npy')\n",
    "p2g = np.load(\"D:/downloads/HighConfidence_kegg/kegg_300/output_embed__pathway_p2g.out.npy\")\n",
    "len(p2g)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "0d0e377c",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:43.633560Z",
     "start_time": "2022-08-11T11:03:43.616636Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#print(len(mu))\n",
    "#print(len(mu[0]))\n",
    "#mu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "8c748af4",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:44.475911Z",
     "start_time": "2022-08-11T11:03:44.459929Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "#print(len(cov))\n",
    "#print(len(cov[0]))\n",
    "#print(len(cov[0][0]))\n",
    "#cov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "d7419318",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:48.958244Z",
     "start_time": "2022-08-11T11:03:48.946277Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15961\n",
      "300\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([[ 0.14997455,  0.03728954, -0.16986829, ..., -0.721358  ,\n",
       "         0.61343336,  1.6425161 ],\n",
       "       [-0.07542448,  0.14683704, -0.3118971 , ..., -0.17488435,\n",
       "         0.9761588 ,  1.3978577 ],\n",
       "       [-0.16300417, -0.02401887, -0.38765138, ..., -0.4417364 ,\n",
       "         0.21016009,  1.8259343 ],\n",
       "       ...,\n",
       "       [ 0.13495177, -0.1015507 , -0.30962887, ..., -0.00964957,\n",
       "        -0.07726423, -0.07330634],\n",
       "       [ 0.14598647, -0.09112392,  0.08978819, ...,  0.29352707,\n",
       "        -0.11388043,  0.39336783],\n",
       "       [ 0.17215613, -0.00992213, -0.16974814, ...,  0.00513786,\n",
       "        -0.07396954, -0.08642484]], dtype=float32)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(g2g))\n",
    "print(len(g2g[0]))\n",
    "g2g"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "73cb3aa6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:03:49.448938Z",
     "start_time": "2022-08-11T11:03:49.431983Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "320\n",
      "15961\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "array([[13.0048914 , 18.60439682, 10.77054501, ..., 18.91812515,\n",
       "        15.75361538, 18.73526573],\n",
       "       [10.54925823, 11.03357506,  9.80605125, ..., 19.50169754,\n",
       "        13.78735733, 19.94822311],\n",
       "       [ 8.51194   , 10.52040005,  8.52526951, ..., 18.76291275,\n",
       "        17.57646561, 18.84948921],\n",
       "       ...,\n",
       "       [11.01446342, 11.87791634, 10.34681511, ..., 18.742239  ,\n",
       "        17.37133026, 18.41414642],\n",
       "       [ 7.945333  ,  8.79445934,  8.04493618, ..., 19.30413628,\n",
       "        16.78544998, 18.40201569],\n",
       "       [ 7.69208431, 10.33734703,  5.26964617, ..., 18.94577599,\n",
       "        15.73103523, 19.77088165]])"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(p2g))\n",
    "print(len(p2g[0]))\n",
    "p2g"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "51d22c22",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## g2g preprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "65ded80e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:13:43.895245Z",
     "start_time": "2022-08-11T11:13:43.841389Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>290</th>\n",
       "      <th>291</th>\n",
       "      <th>292</th>\n",
       "      <th>293</th>\n",
       "      <th>294</th>\n",
       "      <th>295</th>\n",
       "      <th>296</th>\n",
       "      <th>297</th>\n",
       "      <th>298</th>\n",
       "      <th>299</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ARF5</th>\n",
       "      <td>0.149975</td>\n",
       "      <td>0.037290</td>\n",
       "      <td>-0.169868</td>\n",
       "      <td>-0.157117</td>\n",
       "      <td>-0.114900</td>\n",
       "      <td>-0.002312</td>\n",
       "      <td>-0.230968</td>\n",
       "      <td>0.104046</td>\n",
       "      <td>0.174171</td>\n",
       "      <td>-0.049781</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.089971</td>\n",
       "      <td>-0.254667</td>\n",
       "      <td>-0.280530</td>\n",
       "      <td>-0.800504</td>\n",
       "      <td>-0.506039</td>\n",
       "      <td>0.435302</td>\n",
       "      <td>-1.244120</td>\n",
       "      <td>-0.721358</td>\n",
       "      <td>0.613433</td>\n",
       "      <td>1.642516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAP2</th>\n",
       "      <td>-0.075424</td>\n",
       "      <td>0.146837</td>\n",
       "      <td>-0.311897</td>\n",
       "      <td>0.038846</td>\n",
       "      <td>0.052047</td>\n",
       "      <td>0.086256</td>\n",
       "      <td>-0.088711</td>\n",
       "      <td>0.118783</td>\n",
       "      <td>0.201157</td>\n",
       "      <td>-0.010640</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.160904</td>\n",
       "      <td>-0.095572</td>\n",
       "      <td>-0.097176</td>\n",
       "      <td>-0.553882</td>\n",
       "      <td>-0.411733</td>\n",
       "      <td>0.354265</td>\n",
       "      <td>-1.288214</td>\n",
       "      <td>-0.174884</td>\n",
       "      <td>0.976159</td>\n",
       "      <td>1.397858</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RAB1A</th>\n",
       "      <td>-0.163004</td>\n",
       "      <td>-0.024019</td>\n",
       "      <td>-0.387651</td>\n",
       "      <td>-0.134949</td>\n",
       "      <td>-0.058999</td>\n",
       "      <td>-0.077723</td>\n",
       "      <td>0.170445</td>\n",
       "      <td>0.146167</td>\n",
       "      <td>-0.020136</td>\n",
       "      <td>-0.122569</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.384214</td>\n",
       "      <td>-0.463176</td>\n",
       "      <td>-0.177493</td>\n",
       "      <td>-1.457114</td>\n",
       "      <td>-0.952020</td>\n",
       "      <td>0.987618</td>\n",
       "      <td>-1.650460</td>\n",
       "      <td>-0.441736</td>\n",
       "      <td>0.210160</td>\n",
       "      <td>1.825934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COPE</th>\n",
       "      <td>0.036254</td>\n",
       "      <td>0.097664</td>\n",
       "      <td>-0.322370</td>\n",
       "      <td>-0.105176</td>\n",
       "      <td>0.058348</td>\n",
       "      <td>0.122609</td>\n",
       "      <td>-0.175409</td>\n",
       "      <td>-0.009325</td>\n",
       "      <td>-0.023817</td>\n",
       "      <td>0.050343</td>\n",
       "      <td>...</td>\n",
       "      <td>0.340274</td>\n",
       "      <td>-0.407290</td>\n",
       "      <td>-0.574239</td>\n",
       "      <td>-0.937956</td>\n",
       "      <td>-0.926029</td>\n",
       "      <td>0.167821</td>\n",
       "      <td>-1.245971</td>\n",
       "      <td>-1.114583</td>\n",
       "      <td>-0.066414</td>\n",
       "      <td>1.702096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAP1</th>\n",
       "      <td>0.088679</td>\n",
       "      <td>0.013420</td>\n",
       "      <td>-0.036734</td>\n",
       "      <td>-0.023136</td>\n",
       "      <td>-0.290483</td>\n",
       "      <td>-0.032173</td>\n",
       "      <td>0.000306</td>\n",
       "      <td>0.020311</td>\n",
       "      <td>0.146374</td>\n",
       "      <td>-0.126240</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.122595</td>\n",
       "      <td>-0.262638</td>\n",
       "      <td>0.067512</td>\n",
       "      <td>-0.523326</td>\n",
       "      <td>-0.354528</td>\n",
       "      <td>0.043972</td>\n",
       "      <td>-0.650846</td>\n",
       "      <td>-0.366393</td>\n",
       "      <td>1.010563</td>\n",
       "      <td>1.843736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CFAP77</th>\n",
       "      <td>0.065508</td>\n",
       "      <td>0.194203</td>\n",
       "      <td>-0.143959</td>\n",
       "      <td>-0.112470</td>\n",
       "      <td>-0.156552</td>\n",
       "      <td>-0.376265</td>\n",
       "      <td>0.115548</td>\n",
       "      <td>0.175934</td>\n",
       "      <td>0.032740</td>\n",
       "      <td>0.133873</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.156691</td>\n",
       "      <td>-0.258620</td>\n",
       "      <td>0.017624</td>\n",
       "      <td>-0.253683</td>\n",
       "      <td>-0.211852</td>\n",
       "      <td>0.312278</td>\n",
       "      <td>0.059296</td>\n",
       "      <td>-0.472935</td>\n",
       "      <td>-0.142774</td>\n",
       "      <td>0.186995</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PRR23C</th>\n",
       "      <td>0.164783</td>\n",
       "      <td>-0.031773</td>\n",
       "      <td>-0.071346</td>\n",
       "      <td>0.015850</td>\n",
       "      <td>0.290223</td>\n",
       "      <td>0.036937</td>\n",
       "      <td>0.018359</td>\n",
       "      <td>0.064349</td>\n",
       "      <td>-0.212511</td>\n",
       "      <td>0.095208</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.354655</td>\n",
       "      <td>0.037937</td>\n",
       "      <td>0.037358</td>\n",
       "      <td>-0.078771</td>\n",
       "      <td>-0.155729</td>\n",
       "      <td>-0.284284</td>\n",
       "      <td>0.165660</td>\n",
       "      <td>0.180138</td>\n",
       "      <td>0.054500</td>\n",
       "      <td>0.116083</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IQCF6</th>\n",
       "      <td>0.134952</td>\n",
       "      <td>-0.101551</td>\n",
       "      <td>-0.309629</td>\n",
       "      <td>-0.134761</td>\n",
       "      <td>0.105318</td>\n",
       "      <td>0.002783</td>\n",
       "      <td>-0.087199</td>\n",
       "      <td>0.111974</td>\n",
       "      <td>-0.238812</td>\n",
       "      <td>0.015629</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.026417</td>\n",
       "      <td>-0.120801</td>\n",
       "      <td>0.092237</td>\n",
       "      <td>-0.131247</td>\n",
       "      <td>-0.031017</td>\n",
       "      <td>-0.017382</td>\n",
       "      <td>-0.115295</td>\n",
       "      <td>-0.009650</td>\n",
       "      <td>-0.077264</td>\n",
       "      <td>-0.073306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>S100A7L2</th>\n",
       "      <td>0.145986</td>\n",
       "      <td>-0.091124</td>\n",
       "      <td>0.089788</td>\n",
       "      <td>-0.156575</td>\n",
       "      <td>0.279600</td>\n",
       "      <td>-0.065983</td>\n",
       "      <td>-0.377315</td>\n",
       "      <td>-0.124785</td>\n",
       "      <td>0.022467</td>\n",
       "      <td>0.075294</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.092097</td>\n",
       "      <td>-0.112428</td>\n",
       "      <td>0.240911</td>\n",
       "      <td>-0.247859</td>\n",
       "      <td>-0.166712</td>\n",
       "      <td>-0.137839</td>\n",
       "      <td>-0.055621</td>\n",
       "      <td>0.293527</td>\n",
       "      <td>-0.113880</td>\n",
       "      <td>0.393368</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MTRNR2L3</th>\n",
       "      <td>0.172156</td>\n",
       "      <td>-0.009922</td>\n",
       "      <td>-0.169748</td>\n",
       "      <td>-0.097112</td>\n",
       "      <td>0.090393</td>\n",
       "      <td>0.022789</td>\n",
       "      <td>-0.022820</td>\n",
       "      <td>0.143637</td>\n",
       "      <td>-0.169133</td>\n",
       "      <td>0.058359</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.011298</td>\n",
       "      <td>-0.115990</td>\n",
       "      <td>0.097138</td>\n",
       "      <td>-0.093478</td>\n",
       "      <td>-0.030675</td>\n",
       "      <td>-0.060987</td>\n",
       "      <td>-0.088703</td>\n",
       "      <td>0.005138</td>\n",
       "      <td>-0.073970</td>\n",
       "      <td>-0.086425</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15961 rows × 300 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               0         1         2         3         4         5    \\\n",
       "gene                                                                   \n",
       "ARF5      0.149975  0.037290 -0.169868 -0.157117 -0.114900 -0.002312   \n",
       "ACAP2    -0.075424  0.146837 -0.311897  0.038846  0.052047  0.086256   \n",
       "RAB1A    -0.163004 -0.024019 -0.387651 -0.134949 -0.058999 -0.077723   \n",
       "COPE      0.036254  0.097664 -0.322370 -0.105176  0.058348  0.122609   \n",
       "ACAP1     0.088679  0.013420 -0.036734 -0.023136 -0.290483 -0.032173   \n",
       "...            ...       ...       ...       ...       ...       ...   \n",
       "CFAP77    0.065508  0.194203 -0.143959 -0.112470 -0.156552 -0.376265   \n",
       "PRR23C    0.164783 -0.031773 -0.071346  0.015850  0.290223  0.036937   \n",
       "IQCF6     0.134952 -0.101551 -0.309629 -0.134761  0.105318  0.002783   \n",
       "S100A7L2  0.145986 -0.091124  0.089788 -0.156575  0.279600 -0.065983   \n",
       "MTRNR2L3  0.172156 -0.009922 -0.169748 -0.097112  0.090393  0.022789   \n",
       "\n",
       "               6         7         8         9    ...       290       291  \\\n",
       "gene                                              ...                       \n",
       "ARF5     -0.230968  0.104046  0.174171 -0.049781  ... -0.089971 -0.254667   \n",
       "ACAP2    -0.088711  0.118783  0.201157 -0.010640  ... -0.160904 -0.095572   \n",
       "RAB1A     0.170445  0.146167 -0.020136 -0.122569  ... -0.384214 -0.463176   \n",
       "COPE     -0.175409 -0.009325 -0.023817  0.050343  ...  0.340274 -0.407290   \n",
       "ACAP1     0.000306  0.020311  0.146374 -0.126240  ... -0.122595 -0.262638   \n",
       "...            ...       ...       ...       ...  ...       ...       ...   \n",
       "CFAP77    0.115548  0.175934  0.032740  0.133873  ... -0.156691 -0.258620   \n",
       "PRR23C    0.018359  0.064349 -0.212511  0.095208  ... -0.354655  0.037937   \n",
       "IQCF6    -0.087199  0.111974 -0.238812  0.015629  ... -0.026417 -0.120801   \n",
       "S100A7L2 -0.377315 -0.124785  0.022467  0.075294  ... -0.092097 -0.112428   \n",
       "MTRNR2L3 -0.022820  0.143637 -0.169133  0.058359  ... -0.011298 -0.115990   \n",
       "\n",
       "               292       293       294       295       296       297  \\\n",
       "gene                                                                   \n",
       "ARF5     -0.280530 -0.800504 -0.506039  0.435302 -1.244120 -0.721358   \n",
       "ACAP2    -0.097176 -0.553882 -0.411733  0.354265 -1.288214 -0.174884   \n",
       "RAB1A    -0.177493 -1.457114 -0.952020  0.987618 -1.650460 -0.441736   \n",
       "COPE     -0.574239 -0.937956 -0.926029  0.167821 -1.245971 -1.114583   \n",
       "ACAP1     0.067512 -0.523326 -0.354528  0.043972 -0.650846 -0.366393   \n",
       "...            ...       ...       ...       ...       ...       ...   \n",
       "CFAP77    0.017624 -0.253683 -0.211852  0.312278  0.059296 -0.472935   \n",
       "PRR23C    0.037358 -0.078771 -0.155729 -0.284284  0.165660  0.180138   \n",
       "IQCF6     0.092237 -0.131247 -0.031017 -0.017382 -0.115295 -0.009650   \n",
       "S100A7L2  0.240911 -0.247859 -0.166712 -0.137839 -0.055621  0.293527   \n",
       "MTRNR2L3  0.097138 -0.093478 -0.030675 -0.060987 -0.088703  0.005138   \n",
       "\n",
       "               298       299  \n",
       "gene                          \n",
       "ARF5      0.613433  1.642516  \n",
       "ACAP2     0.976159  1.397858  \n",
       "RAB1A     0.210160  1.825934  \n",
       "COPE     -0.066414  1.702096  \n",
       "ACAP1     1.010563  1.843736  \n",
       "...            ...       ...  \n",
       "CFAP77   -0.142774  0.186995  \n",
       "PRR23C    0.054500  0.116083  \n",
       "IQCF6    -0.077264 -0.073306  \n",
       "S100A7L2 -0.113880  0.393368  \n",
       "MTRNR2L3 -0.073970 -0.086425  \n",
       "\n",
       "[15961 rows x 300 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g2g_dataframe = pd.DataFrame(g2g)\n",
    "g2g_dataframe['gene']=node\n",
    "g2g_dataframe = g2g_dataframe.set_index('gene')\n",
    "g2g_dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "118c4491",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "### extract genes' vector"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "cbe462d1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-10T07:16:45.426464Z",
     "start_time": "2022-08-10T07:16:45.386579Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>290</th>\n",
       "      <th>291</th>\n",
       "      <th>292</th>\n",
       "      <th>293</th>\n",
       "      <th>294</th>\n",
       "      <th>295</th>\n",
       "      <th>296</th>\n",
       "      <th>297</th>\n",
       "      <th>298</th>\n",
       "      <th>299</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>RAB7A</th>\n",
       "      <td>0.022178</td>\n",
       "      <td>-0.055640</td>\n",
       "      <td>-0.201210</td>\n",
       "      <td>-0.015118</td>\n",
       "      <td>-0.188326</td>\n",
       "      <td>-0.015908</td>\n",
       "      <td>-0.000558</td>\n",
       "      <td>0.033286</td>\n",
       "      <td>-0.110532</td>\n",
       "      <td>0.184640</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.467682</td>\n",
       "      <td>-0.182319</td>\n",
       "      <td>-0.664772</td>\n",
       "      <td>-0.781382</td>\n",
       "      <td>-1.150512</td>\n",
       "      <td>0.688804</td>\n",
       "      <td>-1.345718</td>\n",
       "      <td>-0.529433</td>\n",
       "      <td>0.774676</td>\n",
       "      <td>2.158654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VPS29</th>\n",
       "      <td>0.041450</td>\n",
       "      <td>0.001487</td>\n",
       "      <td>0.025648</td>\n",
       "      <td>-0.266313</td>\n",
       "      <td>-0.085632</td>\n",
       "      <td>-0.186372</td>\n",
       "      <td>0.006970</td>\n",
       "      <td>0.165371</td>\n",
       "      <td>0.009342</td>\n",
       "      <td>0.028353</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.570832</td>\n",
       "      <td>-0.261074</td>\n",
       "      <td>0.246742</td>\n",
       "      <td>-0.326312</td>\n",
       "      <td>-1.093452</td>\n",
       "      <td>0.603003</td>\n",
       "      <td>-0.865127</td>\n",
       "      <td>-0.519136</td>\n",
       "      <td>-0.187087</td>\n",
       "      <td>1.857312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VPS35</th>\n",
       "      <td>0.260943</td>\n",
       "      <td>-0.174859</td>\n",
       "      <td>-0.053767</td>\n",
       "      <td>-0.032490</td>\n",
       "      <td>-0.048345</td>\n",
       "      <td>-0.227013</td>\n",
       "      <td>0.329565</td>\n",
       "      <td>0.217826</td>\n",
       "      <td>-0.310671</td>\n",
       "      <td>-0.015732</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.299005</td>\n",
       "      <td>0.092172</td>\n",
       "      <td>-0.595958</td>\n",
       "      <td>-0.791436</td>\n",
       "      <td>-1.035793</td>\n",
       "      <td>0.414756</td>\n",
       "      <td>-1.113792</td>\n",
       "      <td>-0.845958</td>\n",
       "      <td>0.048732</td>\n",
       "      <td>2.167360</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTR2</th>\n",
       "      <td>0.142029</td>\n",
       "      <td>-0.316022</td>\n",
       "      <td>0.119639</td>\n",
       "      <td>0.133224</td>\n",
       "      <td>-0.062078</td>\n",
       "      <td>-0.437041</td>\n",
       "      <td>-0.070897</td>\n",
       "      <td>0.060448</td>\n",
       "      <td>0.196248</td>\n",
       "      <td>0.139881</td>\n",
       "      <td>...</td>\n",
       "      <td>0.345415</td>\n",
       "      <td>0.614687</td>\n",
       "      <td>0.395274</td>\n",
       "      <td>-0.121828</td>\n",
       "      <td>-0.064396</td>\n",
       "      <td>-0.157585</td>\n",
       "      <td>-0.950825</td>\n",
       "      <td>0.111630</td>\n",
       "      <td>1.102215</td>\n",
       "      <td>2.006378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARPC3</th>\n",
       "      <td>0.022142</td>\n",
       "      <td>-0.175853</td>\n",
       "      <td>-0.022178</td>\n",
       "      <td>0.178342</td>\n",
       "      <td>-0.009038</td>\n",
       "      <td>-0.292300</td>\n",
       "      <td>-0.087999</td>\n",
       "      <td>-0.034262</td>\n",
       "      <td>0.009168</td>\n",
       "      <td>0.027508</td>\n",
       "      <td>...</td>\n",
       "      <td>0.223051</td>\n",
       "      <td>0.659796</td>\n",
       "      <td>0.316956</td>\n",
       "      <td>-0.028163</td>\n",
       "      <td>-0.113345</td>\n",
       "      <td>-0.485166</td>\n",
       "      <td>-0.851767</td>\n",
       "      <td>0.269587</td>\n",
       "      <td>0.831210</td>\n",
       "      <td>1.900934</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ARPC4</th>\n",
       "      <td>0.054760</td>\n",
       "      <td>-0.131207</td>\n",
       "      <td>-0.019061</td>\n",
       "      <td>0.171200</td>\n",
       "      <td>-0.037743</td>\n",
       "      <td>-0.248629</td>\n",
       "      <td>-0.111539</td>\n",
       "      <td>-0.085790</td>\n",
       "      <td>0.087632</td>\n",
       "      <td>0.115538</td>\n",
       "      <td>...</td>\n",
       "      <td>0.194346</td>\n",
       "      <td>0.756982</td>\n",
       "      <td>0.392102</td>\n",
       "      <td>0.044142</td>\n",
       "      <td>-0.076646</td>\n",
       "      <td>-0.607057</td>\n",
       "      <td>-0.713026</td>\n",
       "      <td>0.322960</td>\n",
       "      <td>0.937189</td>\n",
       "      <td>1.832824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACTR3</th>\n",
       "      <td>0.113556</td>\n",
       "      <td>-0.141153</td>\n",
       "      <td>-0.084623</td>\n",
       "      <td>0.060836</td>\n",
       "      <td>-0.155838</td>\n",
       "      <td>-0.389953</td>\n",
       "      <td>-0.145922</td>\n",
       "      <td>-0.090755</td>\n",
       "      <td>0.507156</td>\n",
       "      <td>0.273185</td>\n",
       "      <td>...</td>\n",
       "      <td>0.170366</td>\n",
       "      <td>0.685921</td>\n",
       "      <td>0.089276</td>\n",
       "      <td>-0.350696</td>\n",
       "      <td>-0.154139</td>\n",
       "      <td>-0.261650</td>\n",
       "      <td>-1.071062</td>\n",
       "      <td>0.263433</td>\n",
       "      <td>0.706738</td>\n",
       "      <td>2.130516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COMMD3-BMI1</th>\n",
       "      <td>0.189190</td>\n",
       "      <td>-0.007340</td>\n",
       "      <td>-0.028933</td>\n",
       "      <td>-0.066898</td>\n",
       "      <td>0.009092</td>\n",
       "      <td>0.004494</td>\n",
       "      <td>0.121493</td>\n",
       "      <td>-0.119051</td>\n",
       "      <td>-0.045198</td>\n",
       "      <td>-0.120440</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.027518</td>\n",
       "      <td>-0.415906</td>\n",
       "      <td>0.402535</td>\n",
       "      <td>-0.080255</td>\n",
       "      <td>-0.225103</td>\n",
       "      <td>0.509317</td>\n",
       "      <td>0.612932</td>\n",
       "      <td>1.237531</td>\n",
       "      <td>-0.844950</td>\n",
       "      <td>1.914604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPC1</th>\n",
       "      <td>0.102417</td>\n",
       "      <td>-0.139809</td>\n",
       "      <td>-0.173949</td>\n",
       "      <td>-0.056705</td>\n",
       "      <td>-0.178511</td>\n",
       "      <td>0.160102</td>\n",
       "      <td>-0.216418</td>\n",
       "      <td>0.018329</td>\n",
       "      <td>0.034599</td>\n",
       "      <td>0.410413</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.069120</td>\n",
       "      <td>0.006082</td>\n",
       "      <td>-0.606822</td>\n",
       "      <td>-0.586891</td>\n",
       "      <td>-0.367947</td>\n",
       "      <td>-0.180683</td>\n",
       "      <td>0.744067</td>\n",
       "      <td>-0.727627</td>\n",
       "      <td>0.302181</td>\n",
       "      <td>1.789317</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCZ1</th>\n",
       "      <td>-0.019629</td>\n",
       "      <td>-0.079048</td>\n",
       "      <td>-0.056893</td>\n",
       "      <td>-0.048743</td>\n",
       "      <td>-0.181949</td>\n",
       "      <td>0.089275</td>\n",
       "      <td>-0.056253</td>\n",
       "      <td>-0.192317</td>\n",
       "      <td>-0.036125</td>\n",
       "      <td>0.284275</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.277527</td>\n",
       "      <td>-0.352067</td>\n",
       "      <td>-0.532499</td>\n",
       "      <td>-0.813226</td>\n",
       "      <td>-1.035945</td>\n",
       "      <td>0.473256</td>\n",
       "      <td>-1.445041</td>\n",
       "      <td>-0.310756</td>\n",
       "      <td>0.613675</td>\n",
       "      <td>1.312770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACE2</th>\n",
       "      <td>0.048304</td>\n",
       "      <td>-0.015851</td>\n",
       "      <td>0.017926</td>\n",
       "      <td>0.035441</td>\n",
       "      <td>0.113974</td>\n",
       "      <td>0.062308</td>\n",
       "      <td>0.147346</td>\n",
       "      <td>0.120781</td>\n",
       "      <td>-0.015833</td>\n",
       "      <td>-0.036758</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.174573</td>\n",
       "      <td>0.081366</td>\n",
       "      <td>0.010601</td>\n",
       "      <td>-0.012833</td>\n",
       "      <td>0.407395</td>\n",
       "      <td>0.062261</td>\n",
       "      <td>0.465684</td>\n",
       "      <td>-0.405166</td>\n",
       "      <td>1.074125</td>\n",
       "      <td>1.792345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC22</th>\n",
       "      <td>-0.269121</td>\n",
       "      <td>0.008326</td>\n",
       "      <td>-0.211533</td>\n",
       "      <td>0.395432</td>\n",
       "      <td>0.020307</td>\n",
       "      <td>-0.205873</td>\n",
       "      <td>-0.461823</td>\n",
       "      <td>-0.272188</td>\n",
       "      <td>0.176076</td>\n",
       "      <td>0.210872</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.323852</td>\n",
       "      <td>-0.109648</td>\n",
       "      <td>-0.414874</td>\n",
       "      <td>0.554369</td>\n",
       "      <td>-1.358388</td>\n",
       "      <td>0.221877</td>\n",
       "      <td>-0.314125</td>\n",
       "      <td>0.029133</td>\n",
       "      <td>-0.388503</td>\n",
       "      <td>1.797038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ASCC3</th>\n",
       "      <td>-0.042735</td>\n",
       "      <td>0.129697</td>\n",
       "      <td>-0.223488</td>\n",
       "      <td>-0.288109</td>\n",
       "      <td>0.186163</td>\n",
       "      <td>0.176843</td>\n",
       "      <td>-0.391452</td>\n",
       "      <td>0.203736</td>\n",
       "      <td>-0.053131</td>\n",
       "      <td>0.353565</td>\n",
       "      <td>...</td>\n",
       "      <td>0.018436</td>\n",
       "      <td>-0.178167</td>\n",
       "      <td>-0.292689</td>\n",
       "      <td>-0.096113</td>\n",
       "      <td>0.206823</td>\n",
       "      <td>-0.704425</td>\n",
       "      <td>-0.530052</td>\n",
       "      <td>-0.796131</td>\n",
       "      <td>-1.495957</td>\n",
       "      <td>1.307785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TFE3</th>\n",
       "      <td>-0.092435</td>\n",
       "      <td>-0.143547</td>\n",
       "      <td>-0.166067</td>\n",
       "      <td>-0.198689</td>\n",
       "      <td>-0.060836</td>\n",
       "      <td>-0.045929</td>\n",
       "      <td>-0.357048</td>\n",
       "      <td>-0.093568</td>\n",
       "      <td>-0.219920</td>\n",
       "      <td>0.220201</td>\n",
       "      <td>...</td>\n",
       "      <td>0.234257</td>\n",
       "      <td>-0.398365</td>\n",
       "      <td>-0.788910</td>\n",
       "      <td>-0.481058</td>\n",
       "      <td>-0.279985</td>\n",
       "      <td>0.322709</td>\n",
       "      <td>-0.218146</td>\n",
       "      <td>0.567094</td>\n",
       "      <td>-0.314911</td>\n",
       "      <td>2.073266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCZ1B</th>\n",
       "      <td>-0.000380</td>\n",
       "      <td>-0.004028</td>\n",
       "      <td>-0.144361</td>\n",
       "      <td>-0.041813</td>\n",
       "      <td>-0.202777</td>\n",
       "      <td>0.048421</td>\n",
       "      <td>0.053366</td>\n",
       "      <td>-0.056153</td>\n",
       "      <td>-0.010245</td>\n",
       "      <td>0.243640</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.287783</td>\n",
       "      <td>-0.234879</td>\n",
       "      <td>-0.471358</td>\n",
       "      <td>-0.741362</td>\n",
       "      <td>-1.070416</td>\n",
       "      <td>0.446177</td>\n",
       "      <td>-1.315643</td>\n",
       "      <td>-0.287101</td>\n",
       "      <td>0.646794</td>\n",
       "      <td>1.304032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTSL</th>\n",
       "      <td>-0.027649</td>\n",
       "      <td>0.015938</td>\n",
       "      <td>-0.009115</td>\n",
       "      <td>-0.058273</td>\n",
       "      <td>0.046818</td>\n",
       "      <td>-0.085605</td>\n",
       "      <td>-0.285271</td>\n",
       "      <td>-0.097157</td>\n",
       "      <td>-0.250927</td>\n",
       "      <td>0.178632</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.571834</td>\n",
       "      <td>-0.051205</td>\n",
       "      <td>0.253424</td>\n",
       "      <td>-0.686679</td>\n",
       "      <td>-0.573394</td>\n",
       "      <td>-1.053771</td>\n",
       "      <td>0.576637</td>\n",
       "      <td>0.192573</td>\n",
       "      <td>1.024859</td>\n",
       "      <td>1.692528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COMMD7</th>\n",
       "      <td>-0.107583</td>\n",
       "      <td>-0.106210</td>\n",
       "      <td>-0.289197</td>\n",
       "      <td>0.227233</td>\n",
       "      <td>-0.015866</td>\n",
       "      <td>-0.124352</td>\n",
       "      <td>-0.319970</td>\n",
       "      <td>-0.028458</td>\n",
       "      <td>0.014871</td>\n",
       "      <td>0.164544</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.379684</td>\n",
       "      <td>-0.161878</td>\n",
       "      <td>-0.324640</td>\n",
       "      <td>0.588892</td>\n",
       "      <td>-0.986386</td>\n",
       "      <td>-0.006984</td>\n",
       "      <td>-0.164156</td>\n",
       "      <td>0.116548</td>\n",
       "      <td>-0.797563</td>\n",
       "      <td>1.754366</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COMMD5</th>\n",
       "      <td>-0.096358</td>\n",
       "      <td>0.115530</td>\n",
       "      <td>-0.375032</td>\n",
       "      <td>0.290929</td>\n",
       "      <td>-0.138387</td>\n",
       "      <td>-0.110437</td>\n",
       "      <td>-0.262966</td>\n",
       "      <td>-0.169961</td>\n",
       "      <td>0.045324</td>\n",
       "      <td>0.143549</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.659576</td>\n",
       "      <td>-0.393888</td>\n",
       "      <td>-0.034380</td>\n",
       "      <td>0.560881</td>\n",
       "      <td>-0.982826</td>\n",
       "      <td>0.258149</td>\n",
       "      <td>0.039649</td>\n",
       "      <td>-0.202373</td>\n",
       "      <td>-0.847844</td>\n",
       "      <td>1.699037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COMMD8</th>\n",
       "      <td>-0.048052</td>\n",
       "      <td>0.054634</td>\n",
       "      <td>-0.441736</td>\n",
       "      <td>0.540206</td>\n",
       "      <td>0.141549</td>\n",
       "      <td>0.056022</td>\n",
       "      <td>-0.342542</td>\n",
       "      <td>0.003359</td>\n",
       "      <td>0.102818</td>\n",
       "      <td>0.096886</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.302948</td>\n",
       "      <td>-0.320064</td>\n",
       "      <td>-0.262200</td>\n",
       "      <td>0.764584</td>\n",
       "      <td>-1.078746</td>\n",
       "      <td>0.474460</td>\n",
       "      <td>-0.195113</td>\n",
       "      <td>-0.025923</td>\n",
       "      <td>-0.691289</td>\n",
       "      <td>1.634157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COMMD4</th>\n",
       "      <td>-0.148474</td>\n",
       "      <td>0.034495</td>\n",
       "      <td>-0.352822</td>\n",
       "      <td>0.254184</td>\n",
       "      <td>-0.020089</td>\n",
       "      <td>-0.085133</td>\n",
       "      <td>-0.327790</td>\n",
       "      <td>-0.104522</td>\n",
       "      <td>0.089935</td>\n",
       "      <td>0.189952</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.427853</td>\n",
       "      <td>-0.288644</td>\n",
       "      <td>-0.165854</td>\n",
       "      <td>0.616045</td>\n",
       "      <td>-1.131260</td>\n",
       "      <td>0.342977</td>\n",
       "      <td>-0.078308</td>\n",
       "      <td>0.309794</td>\n",
       "      <td>-0.775474</td>\n",
       "      <td>1.610722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COMMD2</th>\n",
       "      <td>-0.057729</td>\n",
       "      <td>-0.005890</td>\n",
       "      <td>-0.253959</td>\n",
       "      <td>0.259676</td>\n",
       "      <td>0.003867</td>\n",
       "      <td>-0.058998</td>\n",
       "      <td>-0.282400</td>\n",
       "      <td>-0.181834</td>\n",
       "      <td>0.085688</td>\n",
       "      <td>0.196531</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.434945</td>\n",
       "      <td>-0.231869</td>\n",
       "      <td>-0.221613</td>\n",
       "      <td>0.466776</td>\n",
       "      <td>-1.014368</td>\n",
       "      <td>0.194330</td>\n",
       "      <td>-0.231512</td>\n",
       "      <td>0.132380</td>\n",
       "      <td>-0.839487</td>\n",
       "      <td>1.579563</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COMMD10</th>\n",
       "      <td>-0.151051</td>\n",
       "      <td>-0.001515</td>\n",
       "      <td>-0.342552</td>\n",
       "      <td>0.247020</td>\n",
       "      <td>-0.059794</td>\n",
       "      <td>-0.082011</td>\n",
       "      <td>-0.292684</td>\n",
       "      <td>-0.112557</td>\n",
       "      <td>0.056145</td>\n",
       "      <td>0.202244</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.437858</td>\n",
       "      <td>-0.320359</td>\n",
       "      <td>-0.285708</td>\n",
       "      <td>0.668075</td>\n",
       "      <td>-1.178857</td>\n",
       "      <td>0.239500</td>\n",
       "      <td>-0.101049</td>\n",
       "      <td>0.206739</td>\n",
       "      <td>-0.741740</td>\n",
       "      <td>1.591390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>COMMD3</th>\n",
       "      <td>-0.198036</td>\n",
       "      <td>-0.064801</td>\n",
       "      <td>-0.287452</td>\n",
       "      <td>0.397981</td>\n",
       "      <td>-0.009842</td>\n",
       "      <td>-0.031833</td>\n",
       "      <td>-0.259872</td>\n",
       "      <td>-0.172535</td>\n",
       "      <td>0.034725</td>\n",
       "      <td>0.127016</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.387370</td>\n",
       "      <td>-0.287917</td>\n",
       "      <td>-0.300656</td>\n",
       "      <td>0.694958</td>\n",
       "      <td>-1.163575</td>\n",
       "      <td>0.242021</td>\n",
       "      <td>-0.112039</td>\n",
       "      <td>0.213943</td>\n",
       "      <td>-0.682828</td>\n",
       "      <td>1.566860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CCDC93</th>\n",
       "      <td>-0.237307</td>\n",
       "      <td>-0.102877</td>\n",
       "      <td>-0.212194</td>\n",
       "      <td>0.110540</td>\n",
       "      <td>0.070669</td>\n",
       "      <td>-0.079632</td>\n",
       "      <td>-0.122693</td>\n",
       "      <td>-0.029911</td>\n",
       "      <td>0.148202</td>\n",
       "      <td>0.162419</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.398475</td>\n",
       "      <td>-0.233517</td>\n",
       "      <td>-0.247848</td>\n",
       "      <td>0.283606</td>\n",
       "      <td>-1.139558</td>\n",
       "      <td>0.377008</td>\n",
       "      <td>-0.430347</td>\n",
       "      <td>-0.082662</td>\n",
       "      <td>-0.383595</td>\n",
       "      <td>1.284702</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WDR91</th>\n",
       "      <td>0.471125</td>\n",
       "      <td>-0.069831</td>\n",
       "      <td>-0.063228</td>\n",
       "      <td>-0.071699</td>\n",
       "      <td>-0.005076</td>\n",
       "      <td>-0.174133</td>\n",
       "      <td>-0.091771</td>\n",
       "      <td>-0.148010</td>\n",
       "      <td>-0.431067</td>\n",
       "      <td>0.033935</td>\n",
       "      <td>...</td>\n",
       "      <td>0.094716</td>\n",
       "      <td>0.141850</td>\n",
       "      <td>-0.561582</td>\n",
       "      <td>0.201056</td>\n",
       "      <td>-0.991566</td>\n",
       "      <td>0.167407</td>\n",
       "      <td>-0.437447</td>\n",
       "      <td>-0.142011</td>\n",
       "      <td>0.325006</td>\n",
       "      <td>1.406244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>WDR81</th>\n",
       "      <td>0.576736</td>\n",
       "      <td>-0.177549</td>\n",
       "      <td>0.021852</td>\n",
       "      <td>-0.017107</td>\n",
       "      <td>0.009778</td>\n",
       "      <td>-0.135101</td>\n",
       "      <td>-0.033010</td>\n",
       "      <td>-0.185875</td>\n",
       "      <td>-0.388532</td>\n",
       "      <td>0.089820</td>\n",
       "      <td>...</td>\n",
       "      <td>0.294476</td>\n",
       "      <td>-0.048937</td>\n",
       "      <td>-0.580978</td>\n",
       "      <td>-0.187318</td>\n",
       "      <td>-1.017555</td>\n",
       "      <td>0.251730</td>\n",
       "      <td>-0.369594</td>\n",
       "      <td>-0.449892</td>\n",
       "      <td>0.519648</td>\n",
       "      <td>1.529836</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPC2</th>\n",
       "      <td>0.237416</td>\n",
       "      <td>-0.113744</td>\n",
       "      <td>-0.126035</td>\n",
       "      <td>-0.165505</td>\n",
       "      <td>-0.063524</td>\n",
       "      <td>0.386771</td>\n",
       "      <td>-0.503873</td>\n",
       "      <td>0.074789</td>\n",
       "      <td>0.061559</td>\n",
       "      <td>0.298099</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.071407</td>\n",
       "      <td>-0.348072</td>\n",
       "      <td>-0.129531</td>\n",
       "      <td>-0.956192</td>\n",
       "      <td>-0.498523</td>\n",
       "      <td>0.104940</td>\n",
       "      <td>0.748576</td>\n",
       "      <td>-1.069126</td>\n",
       "      <td>0.487488</td>\n",
       "      <td>1.448538</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>27 rows × 300 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  0         1         2         3         4         5    \\\n",
       "gene                                                                      \n",
       "RAB7A        0.022178 -0.055640 -0.201210 -0.015118 -0.188326 -0.015908   \n",
       "VPS29        0.041450  0.001487  0.025648 -0.266313 -0.085632 -0.186372   \n",
       "VPS35        0.260943 -0.174859 -0.053767 -0.032490 -0.048345 -0.227013   \n",
       "ACTR2        0.142029 -0.316022  0.119639  0.133224 -0.062078 -0.437041   \n",
       "ARPC3        0.022142 -0.175853 -0.022178  0.178342 -0.009038 -0.292300   \n",
       "ARPC4        0.054760 -0.131207 -0.019061  0.171200 -0.037743 -0.248629   \n",
       "ACTR3        0.113556 -0.141153 -0.084623  0.060836 -0.155838 -0.389953   \n",
       "COMMD3-BMI1  0.189190 -0.007340 -0.028933 -0.066898  0.009092  0.004494   \n",
       "NPC1         0.102417 -0.139809 -0.173949 -0.056705 -0.178511  0.160102   \n",
       "CCZ1        -0.019629 -0.079048 -0.056893 -0.048743 -0.181949  0.089275   \n",
       "ACE2         0.048304 -0.015851  0.017926  0.035441  0.113974  0.062308   \n",
       "CCDC22      -0.269121  0.008326 -0.211533  0.395432  0.020307 -0.205873   \n",
       "ASCC3       -0.042735  0.129697 -0.223488 -0.288109  0.186163  0.176843   \n",
       "TFE3        -0.092435 -0.143547 -0.166067 -0.198689 -0.060836 -0.045929   \n",
       "CCZ1B       -0.000380 -0.004028 -0.144361 -0.041813 -0.202777  0.048421   \n",
       "CTSL        -0.027649  0.015938 -0.009115 -0.058273  0.046818 -0.085605   \n",
       "COMMD7      -0.107583 -0.106210 -0.289197  0.227233 -0.015866 -0.124352   \n",
       "COMMD5      -0.096358  0.115530 -0.375032  0.290929 -0.138387 -0.110437   \n",
       "COMMD8      -0.048052  0.054634 -0.441736  0.540206  0.141549  0.056022   \n",
       "COMMD4      -0.148474  0.034495 -0.352822  0.254184 -0.020089 -0.085133   \n",
       "COMMD2      -0.057729 -0.005890 -0.253959  0.259676  0.003867 -0.058998   \n",
       "COMMD10     -0.151051 -0.001515 -0.342552  0.247020 -0.059794 -0.082011   \n",
       "COMMD3      -0.198036 -0.064801 -0.287452  0.397981 -0.009842 -0.031833   \n",
       "CCDC93      -0.237307 -0.102877 -0.212194  0.110540  0.070669 -0.079632   \n",
       "WDR91        0.471125 -0.069831 -0.063228 -0.071699 -0.005076 -0.174133   \n",
       "WDR81        0.576736 -0.177549  0.021852 -0.017107  0.009778 -0.135101   \n",
       "NPC2         0.237416 -0.113744 -0.126035 -0.165505 -0.063524  0.386771   \n",
       "\n",
       "                  6         7         8         9    ...       290       291  \\\n",
       "gene                                                 ...                       \n",
       "RAB7A       -0.000558  0.033286 -0.110532  0.184640  ... -0.467682 -0.182319   \n",
       "VPS29        0.006970  0.165371  0.009342  0.028353  ... -0.570832 -0.261074   \n",
       "VPS35        0.329565  0.217826 -0.310671 -0.015732  ... -0.299005  0.092172   \n",
       "ACTR2       -0.070897  0.060448  0.196248  0.139881  ...  0.345415  0.614687   \n",
       "ARPC3       -0.087999 -0.034262  0.009168  0.027508  ...  0.223051  0.659796   \n",
       "ARPC4       -0.111539 -0.085790  0.087632  0.115538  ...  0.194346  0.756982   \n",
       "ACTR3       -0.145922 -0.090755  0.507156  0.273185  ...  0.170366  0.685921   \n",
       "COMMD3-BMI1  0.121493 -0.119051 -0.045198 -0.120440  ... -0.027518 -0.415906   \n",
       "NPC1        -0.216418  0.018329  0.034599  0.410413  ... -0.069120  0.006082   \n",
       "CCZ1        -0.056253 -0.192317 -0.036125  0.284275  ... -0.277527 -0.352067   \n",
       "ACE2         0.147346  0.120781 -0.015833 -0.036758  ... -0.174573  0.081366   \n",
       "CCDC22      -0.461823 -0.272188  0.176076  0.210872  ... -0.323852 -0.109648   \n",
       "ASCC3       -0.391452  0.203736 -0.053131  0.353565  ...  0.018436 -0.178167   \n",
       "TFE3        -0.357048 -0.093568 -0.219920  0.220201  ...  0.234257 -0.398365   \n",
       "CCZ1B        0.053366 -0.056153 -0.010245  0.243640  ... -0.287783 -0.234879   \n",
       "CTSL        -0.285271 -0.097157 -0.250927  0.178632  ... -0.571834 -0.051205   \n",
       "COMMD7      -0.319970 -0.028458  0.014871  0.164544  ... -0.379684 -0.161878   \n",
       "COMMD5      -0.262966 -0.169961  0.045324  0.143549  ... -0.659576 -0.393888   \n",
       "COMMD8      -0.342542  0.003359  0.102818  0.096886  ... -0.302948 -0.320064   \n",
       "COMMD4      -0.327790 -0.104522  0.089935  0.189952  ... -0.427853 -0.288644   \n",
       "COMMD2      -0.282400 -0.181834  0.085688  0.196531  ... -0.434945 -0.231869   \n",
       "COMMD10     -0.292684 -0.112557  0.056145  0.202244  ... -0.437858 -0.320359   \n",
       "COMMD3      -0.259872 -0.172535  0.034725  0.127016  ... -0.387370 -0.287917   \n",
       "CCDC93      -0.122693 -0.029911  0.148202  0.162419  ... -0.398475 -0.233517   \n",
       "WDR91       -0.091771 -0.148010 -0.431067  0.033935  ...  0.094716  0.141850   \n",
       "WDR81       -0.033010 -0.185875 -0.388532  0.089820  ...  0.294476 -0.048937   \n",
       "NPC2        -0.503873  0.074789  0.061559  0.298099  ... -0.071407 -0.348072   \n",
       "\n",
       "                  292       293       294       295       296       297  \\\n",
       "gene                                                                      \n",
       "RAB7A       -0.664772 -0.781382 -1.150512  0.688804 -1.345718 -0.529433   \n",
       "VPS29        0.246742 -0.326312 -1.093452  0.603003 -0.865127 -0.519136   \n",
       "VPS35       -0.595958 -0.791436 -1.035793  0.414756 -1.113792 -0.845958   \n",
       "ACTR2        0.395274 -0.121828 -0.064396 -0.157585 -0.950825  0.111630   \n",
       "ARPC3        0.316956 -0.028163 -0.113345 -0.485166 -0.851767  0.269587   \n",
       "ARPC4        0.392102  0.044142 -0.076646 -0.607057 -0.713026  0.322960   \n",
       "ACTR3        0.089276 -0.350696 -0.154139 -0.261650 -1.071062  0.263433   \n",
       "COMMD3-BMI1  0.402535 -0.080255 -0.225103  0.509317  0.612932  1.237531   \n",
       "NPC1        -0.606822 -0.586891 -0.367947 -0.180683  0.744067 -0.727627   \n",
       "CCZ1        -0.532499 -0.813226 -1.035945  0.473256 -1.445041 -0.310756   \n",
       "ACE2         0.010601 -0.012833  0.407395  0.062261  0.465684 -0.405166   \n",
       "CCDC22      -0.414874  0.554369 -1.358388  0.221877 -0.314125  0.029133   \n",
       "ASCC3       -0.292689 -0.096113  0.206823 -0.704425 -0.530052 -0.796131   \n",
       "TFE3        -0.788910 -0.481058 -0.279985  0.322709 -0.218146  0.567094   \n",
       "CCZ1B       -0.471358 -0.741362 -1.070416  0.446177 -1.315643 -0.287101   \n",
       "CTSL         0.253424 -0.686679 -0.573394 -1.053771  0.576637  0.192573   \n",
       "COMMD7      -0.324640  0.588892 -0.986386 -0.006984 -0.164156  0.116548   \n",
       "COMMD5      -0.034380  0.560881 -0.982826  0.258149  0.039649 -0.202373   \n",
       "COMMD8      -0.262200  0.764584 -1.078746  0.474460 -0.195113 -0.025923   \n",
       "COMMD4      -0.165854  0.616045 -1.131260  0.342977 -0.078308  0.309794   \n",
       "COMMD2      -0.221613  0.466776 -1.014368  0.194330 -0.231512  0.132380   \n",
       "COMMD10     -0.285708  0.668075 -1.178857  0.239500 -0.101049  0.206739   \n",
       "COMMD3      -0.300656  0.694958 -1.163575  0.242021 -0.112039  0.213943   \n",
       "CCDC93      -0.247848  0.283606 -1.139558  0.377008 -0.430347 -0.082662   \n",
       "WDR91       -0.561582  0.201056 -0.991566  0.167407 -0.437447 -0.142011   \n",
       "WDR81       -0.580978 -0.187318 -1.017555  0.251730 -0.369594 -0.449892   \n",
       "NPC2        -0.129531 -0.956192 -0.498523  0.104940  0.748576 -1.069126   \n",
       "\n",
       "                  298       299  \n",
       "gene                             \n",
       "RAB7A        0.774676  2.158654  \n",
       "VPS29       -0.187087  1.857312  \n",
       "VPS35        0.048732  2.167360  \n",
       "ACTR2        1.102215  2.006378  \n",
       "ARPC3        0.831210  1.900934  \n",
       "ARPC4        0.937189  1.832824  \n",
       "ACTR3        0.706738  2.130516  \n",
       "COMMD3-BMI1 -0.844950  1.914604  \n",
       "NPC1         0.302181  1.789317  \n",
       "CCZ1         0.613675  1.312770  \n",
       "ACE2         1.074125  1.792345  \n",
       "CCDC22      -0.388503  1.797038  \n",
       "ASCC3       -1.495957  1.307785  \n",
       "TFE3        -0.314911  2.073266  \n",
       "CCZ1B        0.646794  1.304032  \n",
       "CTSL         1.024859  1.692528  \n",
       "COMMD7      -0.797563  1.754366  \n",
       "COMMD5      -0.847844  1.699037  \n",
       "COMMD8      -0.691289  1.634157  \n",
       "COMMD4      -0.775474  1.610722  \n",
       "COMMD2      -0.839487  1.579563  \n",
       "COMMD10     -0.741740  1.591390  \n",
       "COMMD3      -0.682828  1.566860  \n",
       "CCDC93      -0.383595  1.284702  \n",
       "WDR91        0.325006  1.406244  \n",
       "WDR81        0.519648  1.529836  \n",
       "NPC2         0.487488  1.448538  \n",
       "\n",
       "[27 rows x 300 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g2g_dataframe_disease_pathway = g2g_dataframe[g2g_dataframe.index.isin(ace2)]\n",
    "g2g_dataframe_disease_pathway"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d3fd1db5",
   "metadata": {
    "heading_collapsed": true,
    "hidden": true
   },
   "source": [
    "## p2d preprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "6ea9b5a4",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:04:06.666246Z",
     "start_time": "2022-08-11T11:04:06.613386Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>15951</th>\n",
       "      <th>15952</th>\n",
       "      <th>15953</th>\n",
       "      <th>15954</th>\n",
       "      <th>15955</th>\n",
       "      <th>15956</th>\n",
       "      <th>15957</th>\n",
       "      <th>15958</th>\n",
       "      <th>15959</th>\n",
       "      <th>15960</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_set_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Ras signaling pathway</th>\n",
       "      <td>13.004891</td>\n",
       "      <td>18.604397</td>\n",
       "      <td>10.770545</td>\n",
       "      <td>11.742773</td>\n",
       "      <td>12.278329</td>\n",
       "      <td>13.361037</td>\n",
       "      <td>14.020433</td>\n",
       "      <td>12.585869</td>\n",
       "      <td>9.968098</td>\n",
       "      <td>14.672712</td>\n",
       "      <td>...</td>\n",
       "      <td>10.673541</td>\n",
       "      <td>15.083788</td>\n",
       "      <td>18.874718</td>\n",
       "      <td>17.437302</td>\n",
       "      <td>11.635219</td>\n",
       "      <td>14.051967</td>\n",
       "      <td>10.695468</td>\n",
       "      <td>18.918125</td>\n",
       "      <td>15.753615</td>\n",
       "      <td>18.735266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Synthesis and degradation of ketone bodies</th>\n",
       "      <td>10.549258</td>\n",
       "      <td>11.033575</td>\n",
       "      <td>9.806051</td>\n",
       "      <td>10.190495</td>\n",
       "      <td>8.996559</td>\n",
       "      <td>10.496225</td>\n",
       "      <td>10.523148</td>\n",
       "      <td>7.660634</td>\n",
       "      <td>5.964342</td>\n",
       "      <td>10.937921</td>\n",
       "      <td>...</td>\n",
       "      <td>14.675311</td>\n",
       "      <td>15.309574</td>\n",
       "      <td>13.176636</td>\n",
       "      <td>17.839787</td>\n",
       "      <td>18.019396</td>\n",
       "      <td>18.009800</td>\n",
       "      <td>16.384592</td>\n",
       "      <td>19.501698</td>\n",
       "      <td>13.787357</td>\n",
       "      <td>19.948223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tryptophan metabolism</th>\n",
       "      <td>8.511940</td>\n",
       "      <td>10.520400</td>\n",
       "      <td>8.525270</td>\n",
       "      <td>10.302014</td>\n",
       "      <td>7.914603</td>\n",
       "      <td>11.500504</td>\n",
       "      <td>10.822841</td>\n",
       "      <td>8.972867</td>\n",
       "      <td>5.907670</td>\n",
       "      <td>10.473205</td>\n",
       "      <td>...</td>\n",
       "      <td>14.521247</td>\n",
       "      <td>14.120238</td>\n",
       "      <td>15.393194</td>\n",
       "      <td>16.794025</td>\n",
       "      <td>18.552992</td>\n",
       "      <td>19.267550</td>\n",
       "      <td>17.295149</td>\n",
       "      <td>18.762913</td>\n",
       "      <td>17.576466</td>\n",
       "      <td>18.849489</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cellular senescence</th>\n",
       "      <td>10.276177</td>\n",
       "      <td>11.345402</td>\n",
       "      <td>9.506522</td>\n",
       "      <td>10.369909</td>\n",
       "      <td>9.442755</td>\n",
       "      <td>11.060408</td>\n",
       "      <td>12.644639</td>\n",
       "      <td>7.495144</td>\n",
       "      <td>5.618484</td>\n",
       "      <td>11.946845</td>\n",
       "      <td>...</td>\n",
       "      <td>13.915819</td>\n",
       "      <td>16.762886</td>\n",
       "      <td>15.561532</td>\n",
       "      <td>18.979267</td>\n",
       "      <td>15.641437</td>\n",
       "      <td>17.741268</td>\n",
       "      <td>15.099967</td>\n",
       "      <td>20.585846</td>\n",
       "      <td>16.619169</td>\n",
       "      <td>20.295670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Melanoma</th>\n",
       "      <td>10.338791</td>\n",
       "      <td>11.616563</td>\n",
       "      <td>7.900808</td>\n",
       "      <td>9.676971</td>\n",
       "      <td>11.094644</td>\n",
       "      <td>9.397411</td>\n",
       "      <td>11.215753</td>\n",
       "      <td>7.710507</td>\n",
       "      <td>4.410573</td>\n",
       "      <td>9.654911</td>\n",
       "      <td>...</td>\n",
       "      <td>14.176823</td>\n",
       "      <td>15.460918</td>\n",
       "      <td>18.076948</td>\n",
       "      <td>17.905724</td>\n",
       "      <td>19.393864</td>\n",
       "      <td>17.818716</td>\n",
       "      <td>17.743242</td>\n",
       "      <td>19.693615</td>\n",
       "      <td>16.468264</td>\n",
       "      <td>20.080921</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fatty acid elongation</th>\n",
       "      <td>8.012397</td>\n",
       "      <td>9.935248</td>\n",
       "      <td>5.185310</td>\n",
       "      <td>8.025328</td>\n",
       "      <td>7.645795</td>\n",
       "      <td>8.087513</td>\n",
       "      <td>9.565404</td>\n",
       "      <td>7.208513</td>\n",
       "      <td>4.885524</td>\n",
       "      <td>8.757915</td>\n",
       "      <td>...</td>\n",
       "      <td>14.553044</td>\n",
       "      <td>14.215632</td>\n",
       "      <td>12.517259</td>\n",
       "      <td>17.370972</td>\n",
       "      <td>18.435896</td>\n",
       "      <td>16.272781</td>\n",
       "      <td>16.137571</td>\n",
       "      <td>18.849882</td>\n",
       "      <td>15.194713</td>\n",
       "      <td>19.197926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mineral absorption</th>\n",
       "      <td>7.773013</td>\n",
       "      <td>10.744399</td>\n",
       "      <td>6.619865</td>\n",
       "      <td>8.377044</td>\n",
       "      <td>10.248653</td>\n",
       "      <td>8.882352</td>\n",
       "      <td>9.186312</td>\n",
       "      <td>6.799828</td>\n",
       "      <td>6.235319</td>\n",
       "      <td>8.192952</td>\n",
       "      <td>...</td>\n",
       "      <td>13.381736</td>\n",
       "      <td>14.050641</td>\n",
       "      <td>13.347853</td>\n",
       "      <td>16.770800</td>\n",
       "      <td>19.046362</td>\n",
       "      <td>18.393661</td>\n",
       "      <td>17.909298</td>\n",
       "      <td>18.988417</td>\n",
       "      <td>17.036463</td>\n",
       "      <td>19.364380</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Legionellosis</th>\n",
       "      <td>11.014463</td>\n",
       "      <td>11.877916</td>\n",
       "      <td>10.346815</td>\n",
       "      <td>10.681679</td>\n",
       "      <td>11.958755</td>\n",
       "      <td>9.379041</td>\n",
       "      <td>11.521346</td>\n",
       "      <td>9.096294</td>\n",
       "      <td>6.651197</td>\n",
       "      <td>9.080625</td>\n",
       "      <td>...</td>\n",
       "      <td>14.327749</td>\n",
       "      <td>16.525978</td>\n",
       "      <td>17.815496</td>\n",
       "      <td>18.115089</td>\n",
       "      <td>19.703150</td>\n",
       "      <td>17.968981</td>\n",
       "      <td>18.849733</td>\n",
       "      <td>18.742239</td>\n",
       "      <td>17.371330</td>\n",
       "      <td>18.414146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vascular smooth muscle contraction</th>\n",
       "      <td>7.945333</td>\n",
       "      <td>8.794459</td>\n",
       "      <td>8.044936</td>\n",
       "      <td>8.845881</td>\n",
       "      <td>7.499012</td>\n",
       "      <td>9.197755</td>\n",
       "      <td>10.165154</td>\n",
       "      <td>6.610901</td>\n",
       "      <td>5.228885</td>\n",
       "      <td>8.893410</td>\n",
       "      <td>...</td>\n",
       "      <td>14.701588</td>\n",
       "      <td>15.297658</td>\n",
       "      <td>14.953139</td>\n",
       "      <td>17.378485</td>\n",
       "      <td>18.899155</td>\n",
       "      <td>16.883688</td>\n",
       "      <td>17.594791</td>\n",
       "      <td>19.304136</td>\n",
       "      <td>16.785450</td>\n",
       "      <td>18.402016</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nicotine addiction</th>\n",
       "      <td>7.692084</td>\n",
       "      <td>10.337347</td>\n",
       "      <td>5.269646</td>\n",
       "      <td>7.934579</td>\n",
       "      <td>8.343276</td>\n",
       "      <td>8.747339</td>\n",
       "      <td>9.423052</td>\n",
       "      <td>9.130812</td>\n",
       "      <td>7.127965</td>\n",
       "      <td>9.038717</td>\n",
       "      <td>...</td>\n",
       "      <td>14.560844</td>\n",
       "      <td>17.099154</td>\n",
       "      <td>10.675890</td>\n",
       "      <td>19.353052</td>\n",
       "      <td>18.349897</td>\n",
       "      <td>17.122292</td>\n",
       "      <td>18.314499</td>\n",
       "      <td>18.945776</td>\n",
       "      <td>15.731035</td>\n",
       "      <td>19.770882</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320 rows × 15961 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                0          1          2      \\\n",
       "gene_set_name                                                                 \n",
       "Ras signaling pathway                       13.004891  18.604397  10.770545   \n",
       "Synthesis and degradation of ketone bodies  10.549258  11.033575   9.806051   \n",
       "Tryptophan metabolism                        8.511940  10.520400   8.525270   \n",
       "Cellular senescence                         10.276177  11.345402   9.506522   \n",
       "Melanoma                                    10.338791  11.616563   7.900808   \n",
       "...                                               ...        ...        ...   \n",
       "Fatty acid elongation                        8.012397   9.935248   5.185310   \n",
       "Mineral absorption                           7.773013  10.744399   6.619865   \n",
       "Legionellosis                               11.014463  11.877916  10.346815   \n",
       "Vascular smooth muscle contraction           7.945333   8.794459   8.044936   \n",
       "Nicotine addiction                           7.692084  10.337347   5.269646   \n",
       "\n",
       "                                                3          4          5      \\\n",
       "gene_set_name                                                                 \n",
       "Ras signaling pathway                       11.742773  12.278329  13.361037   \n",
       "Synthesis and degradation of ketone bodies  10.190495   8.996559  10.496225   \n",
       "Tryptophan metabolism                       10.302014   7.914603  11.500504   \n",
       "Cellular senescence                         10.369909   9.442755  11.060408   \n",
       "Melanoma                                     9.676971  11.094644   9.397411   \n",
       "...                                               ...        ...        ...   \n",
       "Fatty acid elongation                        8.025328   7.645795   8.087513   \n",
       "Mineral absorption                           8.377044  10.248653   8.882352   \n",
       "Legionellosis                               10.681679  11.958755   9.379041   \n",
       "Vascular smooth muscle contraction           8.845881   7.499012   9.197755   \n",
       "Nicotine addiction                           7.934579   8.343276   8.747339   \n",
       "\n",
       "                                                6          7         8      \\\n",
       "gene_set_name                                                                \n",
       "Ras signaling pathway                       14.020433  12.585869  9.968098   \n",
       "Synthesis and degradation of ketone bodies  10.523148   7.660634  5.964342   \n",
       "Tryptophan metabolism                       10.822841   8.972867  5.907670   \n",
       "Cellular senescence                         12.644639   7.495144  5.618484   \n",
       "Melanoma                                    11.215753   7.710507  4.410573   \n",
       "...                                               ...        ...       ...   \n",
       "Fatty acid elongation                        9.565404   7.208513  4.885524   \n",
       "Mineral absorption                           9.186312   6.799828  6.235319   \n",
       "Legionellosis                               11.521346   9.096294  6.651197   \n",
       "Vascular smooth muscle contraction          10.165154   6.610901  5.228885   \n",
       "Nicotine addiction                           9.423052   9.130812  7.127965   \n",
       "\n",
       "                                                9      ...      15951  \\\n",
       "gene_set_name                                          ...              \n",
       "Ras signaling pathway                       14.672712  ...  10.673541   \n",
       "Synthesis and degradation of ketone bodies  10.937921  ...  14.675311   \n",
       "Tryptophan metabolism                       10.473205  ...  14.521247   \n",
       "Cellular senescence                         11.946845  ...  13.915819   \n",
       "Melanoma                                     9.654911  ...  14.176823   \n",
       "...                                               ...  ...        ...   \n",
       "Fatty acid elongation                        8.757915  ...  14.553044   \n",
       "Mineral absorption                           8.192952  ...  13.381736   \n",
       "Legionellosis                                9.080625  ...  14.327749   \n",
       "Vascular smooth muscle contraction           8.893410  ...  14.701588   \n",
       "Nicotine addiction                           9.038717  ...  14.560844   \n",
       "\n",
       "                                                15952      15953      15954  \\\n",
       "gene_set_name                                                                 \n",
       "Ras signaling pathway                       15.083788  18.874718  17.437302   \n",
       "Synthesis and degradation of ketone bodies  15.309574  13.176636  17.839787   \n",
       "Tryptophan metabolism                       14.120238  15.393194  16.794025   \n",
       "Cellular senescence                         16.762886  15.561532  18.979267   \n",
       "Melanoma                                    15.460918  18.076948  17.905724   \n",
       "...                                               ...        ...        ...   \n",
       "Fatty acid elongation                       14.215632  12.517259  17.370972   \n",
       "Mineral absorption                          14.050641  13.347853  16.770800   \n",
       "Legionellosis                               16.525978  17.815496  18.115089   \n",
       "Vascular smooth muscle contraction          15.297658  14.953139  17.378485   \n",
       "Nicotine addiction                          17.099154  10.675890  19.353052   \n",
       "\n",
       "                                                15955      15956      15957  \\\n",
       "gene_set_name                                                                 \n",
       "Ras signaling pathway                       11.635219  14.051967  10.695468   \n",
       "Synthesis and degradation of ketone bodies  18.019396  18.009800  16.384592   \n",
       "Tryptophan metabolism                       18.552992  19.267550  17.295149   \n",
       "Cellular senescence                         15.641437  17.741268  15.099967   \n",
       "Melanoma                                    19.393864  17.818716  17.743242   \n",
       "...                                               ...        ...        ...   \n",
       "Fatty acid elongation                       18.435896  16.272781  16.137571   \n",
       "Mineral absorption                          19.046362  18.393661  17.909298   \n",
       "Legionellosis                               19.703150  17.968981  18.849733   \n",
       "Vascular smooth muscle contraction          18.899155  16.883688  17.594791   \n",
       "Nicotine addiction                          18.349897  17.122292  18.314499   \n",
       "\n",
       "                                                15958      15959      15960  \n",
       "gene_set_name                                                                \n",
       "Ras signaling pathway                       18.918125  15.753615  18.735266  \n",
       "Synthesis and degradation of ketone bodies  19.501698  13.787357  19.948223  \n",
       "Tryptophan metabolism                       18.762913  17.576466  18.849489  \n",
       "Cellular senescence                         20.585846  16.619169  20.295670  \n",
       "Melanoma                                    19.693615  16.468264  20.080921  \n",
       "...                                               ...        ...        ...  \n",
       "Fatty acid elongation                       18.849882  15.194713  19.197926  \n",
       "Mineral absorption                          18.988417  17.036463  19.364380  \n",
       "Legionellosis                               18.742239  17.371330  18.414146  \n",
       "Vascular smooth muscle contraction          19.304136  16.785450  18.402016  \n",
       "Nicotine addiction                          18.945776  15.731035  19.770882  \n",
       "\n",
       "[320 rows x 15961 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p2g_df = pd.DataFrame(p2g)\n",
    "p2g_df['gene_set_name']=kegg_gene_set_name\n",
    "#p2g.set_index(reactome_name,inplace=True)\n",
    "p2g_df = p2g_df.set_index('gene_set_name')\n",
    "p2g_df"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cfd8d0a8",
   "metadata": {},
   "source": [
    "# analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ec83560",
   "metadata": {},
   "source": [
    "## genes' vector mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "9c6fe615",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:14:23.267604Z",
     "start_time": "2022-08-11T11:14:23.259600Z"
    }
   },
   "outputs": [],
   "source": [
    "def gene_set_mean(g2g_dataframe_disease_pathway):\n",
    "    df = copy.deepcopy(g2g_dataframe_disease_pathway)\n",
    "    value = df.mean()\n",
    "    return value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "76baecba",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-10T07:17:00.274307Z",
     "start_time": "2022-08-10T07:17:00.257325Z"
    }
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0      0.029100\n",
       "1     -0.061728\n",
       "2     -0.146456\n",
       "3      0.073140\n",
       "4     -0.034127\n",
       "         ...   \n",
       "295    0.108791\n",
       "296   -0.358013\n",
       "297   -0.089628\n",
       "298    0.014937\n",
       "299    1.709003\n",
       "Length: 300, dtype: float32"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_set_mean(g2g_dataframe_disease_pathway)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "2f1e071b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T15:00:57.781621Z",
     "start_time": "2022-08-11T15:00:57.770650Z"
    }
   },
   "outputs": [],
   "source": [
    "def median(g2g_dataframe_disease_pathway):\n",
    "    df = copy.deepcopy(g2g_dataframe_disease_pathway)\n",
    "    value = df.median()\n",
    "    return value"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d53cf9a2",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "be20c9ff",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-13T11:52:21.825508Z",
     "start_time": "2022-06-13T11:52:21.794637Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>290</th>\n",
       "      <th>291</th>\n",
       "      <th>292</th>\n",
       "      <th>293</th>\n",
       "      <th>294</th>\n",
       "      <th>295</th>\n",
       "      <th>296</th>\n",
       "      <th>297</th>\n",
       "      <th>298</th>\n",
       "      <th>299</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>PSMD4</th>\n",
       "      <td>-0.137230</td>\n",
       "      <td>0.119267</td>\n",
       "      <td>0.172397</td>\n",
       "      <td>0.098050</td>\n",
       "      <td>0.062651</td>\n",
       "      <td>-0.121894</td>\n",
       "      <td>-0.041988</td>\n",
       "      <td>0.131091</td>\n",
       "      <td>-0.025261</td>\n",
       "      <td>-0.068476</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.025660</td>\n",
       "      <td>-0.724149</td>\n",
       "      <td>0.006923</td>\n",
       "      <td>0.708591</td>\n",
       "      <td>-0.998344</td>\n",
       "      <td>0.146174</td>\n",
       "      <td>-0.413383</td>\n",
       "      <td>0.036258</td>\n",
       "      <td>-0.589303</td>\n",
       "      <td>2.307654</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RHOA</th>\n",
       "      <td>0.174755</td>\n",
       "      <td>-0.122616</td>\n",
       "      <td>0.024465</td>\n",
       "      <td>0.222977</td>\n",
       "      <td>0.153304</td>\n",
       "      <td>-0.385418</td>\n",
       "      <td>0.049415</td>\n",
       "      <td>0.078559</td>\n",
       "      <td>-0.111955</td>\n",
       "      <td>0.047027</td>\n",
       "      <td>...</td>\n",
       "      <td>0.471123</td>\n",
       "      <td>0.505535</td>\n",
       "      <td>0.608676</td>\n",
       "      <td>0.334893</td>\n",
       "      <td>0.328151</td>\n",
       "      <td>0.077682</td>\n",
       "      <td>-0.661063</td>\n",
       "      <td>0.324358</td>\n",
       "      <td>1.374497</td>\n",
       "      <td>2.670882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RBX1</th>\n",
       "      <td>0.154529</td>\n",
       "      <td>0.135251</td>\n",
       "      <td>-0.003179</td>\n",
       "      <td>0.009166</td>\n",
       "      <td>0.140370</td>\n",
       "      <td>-0.129565</td>\n",
       "      <td>-0.027169</td>\n",
       "      <td>-0.185461</td>\n",
       "      <td>0.024644</td>\n",
       "      <td>-0.105711</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.056876</td>\n",
       "      <td>-0.347491</td>\n",
       "      <td>-0.211646</td>\n",
       "      <td>0.549933</td>\n",
       "      <td>-0.866758</td>\n",
       "      <td>0.302497</td>\n",
       "      <td>0.055007</td>\n",
       "      <td>0.672922</td>\n",
       "      <td>-1.182373</td>\n",
       "      <td>2.316441</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PABPC1</th>\n",
       "      <td>0.002531</td>\n",
       "      <td>0.002936</td>\n",
       "      <td>0.047458</td>\n",
       "      <td>0.104582</td>\n",
       "      <td>0.086855</td>\n",
       "      <td>-0.205164</td>\n",
       "      <td>-0.216906</td>\n",
       "      <td>-0.290588</td>\n",
       "      <td>-0.120903</td>\n",
       "      <td>0.161402</td>\n",
       "      <td>...</td>\n",
       "      <td>0.190838</td>\n",
       "      <td>0.140339</td>\n",
       "      <td>-0.305139</td>\n",
       "      <td>-0.109511</td>\n",
       "      <td>0.146524</td>\n",
       "      <td>-0.953888</td>\n",
       "      <td>-0.725125</td>\n",
       "      <td>-0.264740</td>\n",
       "      <td>-1.353784</td>\n",
       "      <td>2.190597</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPS25</th>\n",
       "      <td>0.106036</td>\n",
       "      <td>0.006605</td>\n",
       "      <td>-0.114037</td>\n",
       "      <td>-0.037651</td>\n",
       "      <td>0.070181</td>\n",
       "      <td>-0.117366</td>\n",
       "      <td>-0.110169</td>\n",
       "      <td>-0.006673</td>\n",
       "      <td>-0.090432</td>\n",
       "      <td>0.179433</td>\n",
       "      <td>...</td>\n",
       "      <td>0.222499</td>\n",
       "      <td>-0.271341</td>\n",
       "      <td>0.471220</td>\n",
       "      <td>0.886219</td>\n",
       "      <td>-0.231662</td>\n",
       "      <td>-0.898924</td>\n",
       "      <td>-0.316474</td>\n",
       "      <td>-1.357710</td>\n",
       "      <td>-1.146208</td>\n",
       "      <td>0.878045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HECTD1</th>\n",
       "      <td>-0.018660</td>\n",
       "      <td>-0.210056</td>\n",
       "      <td>0.082394</td>\n",
       "      <td>-0.212041</td>\n",
       "      <td>0.231722</td>\n",
       "      <td>0.268953</td>\n",
       "      <td>0.074765</td>\n",
       "      <td>0.193019</td>\n",
       "      <td>0.160030</td>\n",
       "      <td>0.008409</td>\n",
       "      <td>...</td>\n",
       "      <td>0.013655</td>\n",
       "      <td>-0.085049</td>\n",
       "      <td>-0.303765</td>\n",
       "      <td>0.167886</td>\n",
       "      <td>-0.508031</td>\n",
       "      <td>-0.126034</td>\n",
       "      <td>-0.326815</td>\n",
       "      <td>0.276336</td>\n",
       "      <td>-1.140766</td>\n",
       "      <td>1.516213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC35B1</th>\n",
       "      <td>0.282142</td>\n",
       "      <td>-0.116001</td>\n",
       "      <td>0.118198</td>\n",
       "      <td>0.131503</td>\n",
       "      <td>0.070781</td>\n",
       "      <td>-0.200342</td>\n",
       "      <td>-0.166209</td>\n",
       "      <td>-0.211428</td>\n",
       "      <td>-0.008591</td>\n",
       "      <td>0.458990</td>\n",
       "      <td>...</td>\n",
       "      <td>0.459590</td>\n",
       "      <td>-0.630945</td>\n",
       "      <td>-0.094847</td>\n",
       "      <td>-0.832305</td>\n",
       "      <td>-0.440390</td>\n",
       "      <td>0.031554</td>\n",
       "      <td>0.180365</td>\n",
       "      <td>-1.315206</td>\n",
       "      <td>0.033524</td>\n",
       "      <td>1.388198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NDUFAF2</th>\n",
       "      <td>-0.130198</td>\n",
       "      <td>0.212899</td>\n",
       "      <td>0.064541</td>\n",
       "      <td>-0.119938</td>\n",
       "      <td>0.042641</td>\n",
       "      <td>-0.087553</td>\n",
       "      <td>-0.206809</td>\n",
       "      <td>-0.042760</td>\n",
       "      <td>-0.144668</td>\n",
       "      <td>0.088668</td>\n",
       "      <td>...</td>\n",
       "      <td>1.061115</td>\n",
       "      <td>0.158037</td>\n",
       "      <td>-0.330854</td>\n",
       "      <td>0.627966</td>\n",
       "      <td>-0.352194</td>\n",
       "      <td>0.434270</td>\n",
       "      <td>0.548585</td>\n",
       "      <td>-1.400304</td>\n",
       "      <td>-0.621330</td>\n",
       "      <td>0.872879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC30A9</th>\n",
       "      <td>-0.085075</td>\n",
       "      <td>-0.021696</td>\n",
       "      <td>-0.059914</td>\n",
       "      <td>0.272966</td>\n",
       "      <td>-0.317876</td>\n",
       "      <td>0.136142</td>\n",
       "      <td>-0.301886</td>\n",
       "      <td>-0.068124</td>\n",
       "      <td>0.032970</td>\n",
       "      <td>0.096991</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.066986</td>\n",
       "      <td>-0.165720</td>\n",
       "      <td>0.589043</td>\n",
       "      <td>-0.479181</td>\n",
       "      <td>0.262240</td>\n",
       "      <td>0.064210</td>\n",
       "      <td>0.059867</td>\n",
       "      <td>-0.350863</td>\n",
       "      <td>-0.223592</td>\n",
       "      <td>1.011118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NPC2</th>\n",
       "      <td>0.237416</td>\n",
       "      <td>-0.113744</td>\n",
       "      <td>-0.126035</td>\n",
       "      <td>-0.165505</td>\n",
       "      <td>-0.063524</td>\n",
       "      <td>0.386771</td>\n",
       "      <td>-0.503873</td>\n",
       "      <td>0.074789</td>\n",
       "      <td>0.061559</td>\n",
       "      <td>0.298099</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.071407</td>\n",
       "      <td>-0.348072</td>\n",
       "      <td>-0.129531</td>\n",
       "      <td>-0.956192</td>\n",
       "      <td>-0.498523</td>\n",
       "      <td>0.104940</td>\n",
       "      <td>0.748576</td>\n",
       "      <td>-1.069126</td>\n",
       "      <td>0.487488</td>\n",
       "      <td>1.448538</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>72 rows × 300 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              0         1         2         3         4         5         6    \\\n",
       "gene                                                                            \n",
       "PSMD4   -0.137230  0.119267  0.172397  0.098050  0.062651 -0.121894 -0.041988   \n",
       "RHOA     0.174755 -0.122616  0.024465  0.222977  0.153304 -0.385418  0.049415   \n",
       "RBX1     0.154529  0.135251 -0.003179  0.009166  0.140370 -0.129565 -0.027169   \n",
       "PABPC1   0.002531  0.002936  0.047458  0.104582  0.086855 -0.205164 -0.216906   \n",
       "MRPS25   0.106036  0.006605 -0.114037 -0.037651  0.070181 -0.117366 -0.110169   \n",
       "...           ...       ...       ...       ...       ...       ...       ...   \n",
       "HECTD1  -0.018660 -0.210056  0.082394 -0.212041  0.231722  0.268953  0.074765   \n",
       "SLC35B1  0.282142 -0.116001  0.118198  0.131503  0.070781 -0.200342 -0.166209   \n",
       "NDUFAF2 -0.130198  0.212899  0.064541 -0.119938  0.042641 -0.087553 -0.206809   \n",
       "SLC30A9 -0.085075 -0.021696 -0.059914  0.272966 -0.317876  0.136142 -0.301886   \n",
       "NPC2     0.237416 -0.113744 -0.126035 -0.165505 -0.063524  0.386771 -0.503873   \n",
       "\n",
       "              7         8         9    ...       290       291       292  \\\n",
       "gene                                   ...                                 \n",
       "PSMD4    0.131091 -0.025261 -0.068476  ... -0.025660 -0.724149  0.006923   \n",
       "RHOA     0.078559 -0.111955  0.047027  ...  0.471123  0.505535  0.608676   \n",
       "RBX1    -0.185461  0.024644 -0.105711  ... -0.056876 -0.347491 -0.211646   \n",
       "PABPC1  -0.290588 -0.120903  0.161402  ...  0.190838  0.140339 -0.305139   \n",
       "MRPS25  -0.006673 -0.090432  0.179433  ...  0.222499 -0.271341  0.471220   \n",
       "...           ...       ...       ...  ...       ...       ...       ...   \n",
       "HECTD1   0.193019  0.160030  0.008409  ...  0.013655 -0.085049 -0.303765   \n",
       "SLC35B1 -0.211428 -0.008591  0.458990  ...  0.459590 -0.630945 -0.094847   \n",
       "NDUFAF2 -0.042760 -0.144668  0.088668  ...  1.061115  0.158037 -0.330854   \n",
       "SLC30A9 -0.068124  0.032970  0.096991  ... -0.066986 -0.165720  0.589043   \n",
       "NPC2     0.074789  0.061559  0.298099  ... -0.071407 -0.348072 -0.129531   \n",
       "\n",
       "              293       294       295       296       297       298       299  \n",
       "gene                                                                           \n",
       "PSMD4    0.708591 -0.998344  0.146174 -0.413383  0.036258 -0.589303  2.307654  \n",
       "RHOA     0.334893  0.328151  0.077682 -0.661063  0.324358  1.374497  2.670882  \n",
       "RBX1     0.549933 -0.866758  0.302497  0.055007  0.672922 -1.182373  2.316441  \n",
       "PABPC1  -0.109511  0.146524 -0.953888 -0.725125 -0.264740 -1.353784  2.190597  \n",
       "MRPS25   0.886219 -0.231662 -0.898924 -0.316474 -1.357710 -1.146208  0.878045  \n",
       "...           ...       ...       ...       ...       ...       ...       ...  \n",
       "HECTD1   0.167886 -0.508031 -0.126034 -0.326815  0.276336 -1.140766  1.516213  \n",
       "SLC35B1 -0.832305 -0.440390  0.031554  0.180365 -1.315206  0.033524  1.388198  \n",
       "NDUFAF2  0.627966 -0.352194  0.434270  0.548585 -1.400304 -0.621330  0.872879  \n",
       "SLC30A9 -0.479181  0.262240  0.064210  0.059867 -0.350863 -0.223592  1.011118  \n",
       "NPC2    -0.956192 -0.498523  0.104940  0.748576 -1.069126  0.487488  1.448538  \n",
       "\n",
       "[72 rows x 300 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g2g_dataframe_disease_pathway = g2g_dataframe[g2g_dataframe.index.isin(function)]\n",
    "g2g_dataframe_disease_pathway"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5f2a6ad6",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-04-04T16:20:45.473723Z",
     "start_time": "2022-04-04T16:04:55.401720Z"
    },
    "hidden": true
   },
   "source": [
    "### using mu on output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "276fe334",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-13T11:52:57.355602Z",
     "start_time": "2022-06-13T11:52:57.033247Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 1031.69it/s]\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(gene_set_from_cancer_type[0]))\n",
    "    g2g_dataframe_disease_pathway = g2g_dataframe[g2g_dataframe.index.isin(gene_set_from_cancer_type[0][i])]\n",
    "    \n",
    "DiseaseAndPathway_distance = []    \n",
    "    \n",
    "DiseaseAndPathway_mu_distance = []\n",
    "for i in tqdm(range(len(mu))):\n",
    "    mean = gene_set_mean(g2g_dataframe_disease_pathway)\n",
    "    norm = np.linalg.norm(mean - mu[i])\n",
    "    DiseaseAndPathway_mu_distance.append(norm)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "44ec2065",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## using permutation method to get p-value and use it to rank"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "1a6bd7e1",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-13T11:55:29.001615Z",
     "start_time": "2022-06-13T11:53:47.609349Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|████████████████████████████████████████████████████████████████████████████████| 320/320 [01:41<00:00,  3.16it/s]\n"
     ]
    }
   ],
   "source": [
    "node_complement = copy.deepcopy(list(node))\n",
    "for i in node:\n",
    "    if i in COVID_list[0]:\n",
    "        node_complement.remove(i)\n",
    "        \n",
    "pathway_mu_for_random = []\n",
    "for i in tqdm(range(len(mu))):\n",
    "    pathway_mu =[]\n",
    "    for times in range(100):\n",
    "        random_select_gene = random.sample(node_complement,k=len(g2g_dataframe_disease_pathway))\n",
    "        g2g_dataframe_disease_pathway = g2g_dataframe[g2g_dataframe.index.isin(random_select_gene)]\n",
    "        random_select_gene_mean = gene_set_mean(g2g_dataframe_disease_pathway)\n",
    "        mean_rd = gene_set_mean(g2g_dataframe_disease_pathway)\n",
    "        norm_rd = np.linalg.norm(mean_rd - mu[i])\n",
    "        pathway_mu.append(norm_rd)\n",
    "    pathway_mu_for_random.append(pathway_mu)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "id": "0eeb02ad",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-13T12:00:46.976693Z",
     "start_time": "2022-06-13T12:00:45.984699Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|███████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 329.59it/s]\n"
     ]
    }
   ],
   "source": [
    "pathway_disease_z_score = []\n",
    "pathway_disease_p_value = []\n",
    "for number in tqdm(range(len(DiseaseAndPathway_mu_distance))):\n",
    "    mean = np.mean(pathway_mu_for_random)\n",
    "    std = np.std(pathway_mu_for_random)\n",
    "    z_score = (DiseaseAndPathway_mu_distance[number] -mean) / std\n",
    "    pathway_disease_z_score.append(z_score)\n",
    "    pathway_disease_p_value.append(scipy.stats.norm.sf(abs(z_score))*2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "bd51a494",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-06-13T12:01:36.299733Z",
     "start_time": "2022-06-13T12:01:36.288762Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "rank\t1\tE\t0.0019317921350764245\t3.100523\n",
      "rank\t2\tP\t0.00509818875357548\t2.8007634\n",
      "rank\t3\tV\t0.005202553717610762\t2.794217\n",
      "rank\t4\tB\t0.005608387609656068\t2.7698398\n",
      "rank\t5\tT\t0.007257059913241624\t2.684812\n",
      "rank\t6\tT\t0.012747734563074884\t2.49074\n",
      "rank\t7\tS\t0.015143025759945009\t2.4289403\n",
      "rank\t8\tO\t0.01741173262676824\t2.377896\n",
      "rank\t9\tG\t0.019296476849761292\t2.3397534\n"
     ]
    }
   ],
   "source": [
    "rank_z_score_disease_pathway = scipy.stats.rankdata(pathway_disease_z_score)\n",
    "rank_p_value_for_disease_pathway = scipy.stats.rankdata(pathway_disease_p_value)\n",
    "for i in range(1,10):\n",
    "    print('rank',i, reactomedb_list[list(rank_p_value_for_disease_pathway).index(i)][0],\n",
    "          pathway_disease_p_value[list(rank_p_value_for_disease_pathway).index(i)],\n",
    "          pathway_disease_z_score[list(rank_p_value_for_disease_pathway).index(i)],\n",
    "          sep = '\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bbc5d581",
   "metadata": {
    "heading_collapsed": true
   },
   "source": [
    "## Disease and pathway rank using mu distance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "488f916b",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-10T07:18:43.847932Z",
     "start_time": "2022-08-10T07:18:43.588864Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'DiseaseAndPathway_mu_distance' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-33-d4a2187adaf3>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mDiseaseAndPathway_mu_distance\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m: name 'DiseaseAndPathway_mu_distance' is not defined"
     ]
    }
   ],
   "source": [
    "DiseaseAndPathway_mu_distance[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "d2b060d0",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-22T17:44:38.198809Z",
     "start_time": "2022-05-22T17:44:38.183878Z"
    },
    "hidden": true
   },
   "outputs": [],
   "source": [
    "distance_from_geneset_df = copy.deepcopy(DiseaseAndPathway_mu_distance)\n",
    "distance_from_geneset_df = list(map(int,distance_from_geneset_df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "e4d74ae4",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-22T17:44:41.327448Z",
     "start_time": "2022-05-22T17:44:41.240679Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>15953</th>\n",
       "      <th>15954</th>\n",
       "      <th>15955</th>\n",
       "      <th>15956</th>\n",
       "      <th>15957</th>\n",
       "      <th>15958</th>\n",
       "      <th>15959</th>\n",
       "      <th>15960</th>\n",
       "      <th>distance_from_geneset</th>\n",
       "      <th>rank</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gene_set_name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Ras signaling pathway</th>\n",
       "      <td>13.004891</td>\n",
       "      <td>18.604397</td>\n",
       "      <td>10.770545</td>\n",
       "      <td>11.742773</td>\n",
       "      <td>12.278329</td>\n",
       "      <td>13.361037</td>\n",
       "      <td>14.020433</td>\n",
       "      <td>12.585869</td>\n",
       "      <td>9.968098</td>\n",
       "      <td>14.672712</td>\n",
       "      <td>...</td>\n",
       "      <td>18.874718</td>\n",
       "      <td>17.437302</td>\n",
       "      <td>11.635219</td>\n",
       "      <td>14.051967</td>\n",
       "      <td>10.695468</td>\n",
       "      <td>18.918125</td>\n",
       "      <td>15.753615</td>\n",
       "      <td>18.735266</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Synthesis and degradation of ketone bodies</th>\n",
       "      <td>10.549258</td>\n",
       "      <td>11.033575</td>\n",
       "      <td>9.806051</td>\n",
       "      <td>10.190495</td>\n",
       "      <td>8.996559</td>\n",
       "      <td>10.496225</td>\n",
       "      <td>10.523148</td>\n",
       "      <td>7.660634</td>\n",
       "      <td>5.964342</td>\n",
       "      <td>10.937921</td>\n",
       "      <td>...</td>\n",
       "      <td>13.176636</td>\n",
       "      <td>17.839787</td>\n",
       "      <td>18.019396</td>\n",
       "      <td>18.009800</td>\n",
       "      <td>16.384592</td>\n",
       "      <td>19.501698</td>\n",
       "      <td>13.787357</td>\n",
       "      <td>19.948223</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tryptophan metabolism</th>\n",
       "      <td>8.511940</td>\n",
       "      <td>10.520400</td>\n",
       "      <td>8.525270</td>\n",
       "      <td>10.302014</td>\n",
       "      <td>7.914603</td>\n",
       "      <td>11.500504</td>\n",
       "      <td>10.822841</td>\n",
       "      <td>8.972867</td>\n",
       "      <td>5.907670</td>\n",
       "      <td>10.473205</td>\n",
       "      <td>...</td>\n",
       "      <td>15.393194</td>\n",
       "      <td>16.794025</td>\n",
       "      <td>18.552992</td>\n",
       "      <td>19.267550</td>\n",
       "      <td>17.295149</td>\n",
       "      <td>18.762913</td>\n",
       "      <td>17.576466</td>\n",
       "      <td>18.849489</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cellular senescence</th>\n",
       "      <td>10.276177</td>\n",
       "      <td>11.345402</td>\n",
       "      <td>9.506522</td>\n",
       "      <td>10.369909</td>\n",
       "      <td>9.442755</td>\n",
       "      <td>11.060408</td>\n",
       "      <td>12.644639</td>\n",
       "      <td>7.495144</td>\n",
       "      <td>5.618484</td>\n",
       "      <td>11.946845</td>\n",
       "      <td>...</td>\n",
       "      <td>15.561532</td>\n",
       "      <td>18.979267</td>\n",
       "      <td>15.641437</td>\n",
       "      <td>17.741268</td>\n",
       "      <td>15.099967</td>\n",
       "      <td>20.585846</td>\n",
       "      <td>16.619169</td>\n",
       "      <td>20.295670</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Melanoma</th>\n",
       "      <td>10.338791</td>\n",
       "      <td>11.616563</td>\n",
       "      <td>7.900808</td>\n",
       "      <td>9.676971</td>\n",
       "      <td>11.094644</td>\n",
       "      <td>9.397411</td>\n",
       "      <td>11.215753</td>\n",
       "      <td>7.710507</td>\n",
       "      <td>4.410573</td>\n",
       "      <td>9.654911</td>\n",
       "      <td>...</td>\n",
       "      <td>18.076948</td>\n",
       "      <td>17.905724</td>\n",
       "      <td>19.393864</td>\n",
       "      <td>17.818716</td>\n",
       "      <td>17.743242</td>\n",
       "      <td>19.693615</td>\n",
       "      <td>16.468264</td>\n",
       "      <td>20.080921</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Fatty acid elongation</th>\n",
       "      <td>8.012397</td>\n",
       "      <td>9.935248</td>\n",
       "      <td>5.185310</td>\n",
       "      <td>8.025328</td>\n",
       "      <td>7.645795</td>\n",
       "      <td>8.087513</td>\n",
       "      <td>9.565404</td>\n",
       "      <td>7.208513</td>\n",
       "      <td>4.885524</td>\n",
       "      <td>8.757915</td>\n",
       "      <td>...</td>\n",
       "      <td>12.517259</td>\n",
       "      <td>17.370972</td>\n",
       "      <td>18.435896</td>\n",
       "      <td>16.272781</td>\n",
       "      <td>16.137571</td>\n",
       "      <td>18.849882</td>\n",
       "      <td>15.194713</td>\n",
       "      <td>19.197926</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mineral absorption</th>\n",
       "      <td>7.773013</td>\n",
       "      <td>10.744399</td>\n",
       "      <td>6.619865</td>\n",
       "      <td>8.377044</td>\n",
       "      <td>10.248653</td>\n",
       "      <td>8.882352</td>\n",
       "      <td>9.186312</td>\n",
       "      <td>6.799828</td>\n",
       "      <td>6.235319</td>\n",
       "      <td>8.192952</td>\n",
       "      <td>...</td>\n",
       "      <td>13.347853</td>\n",
       "      <td>16.770800</td>\n",
       "      <td>19.046362</td>\n",
       "      <td>18.393661</td>\n",
       "      <td>17.909298</td>\n",
       "      <td>18.988417</td>\n",
       "      <td>17.036463</td>\n",
       "      <td>19.364380</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Legionellosis</th>\n",
       "      <td>11.014463</td>\n",
       "      <td>11.877916</td>\n",
       "      <td>10.346815</td>\n",
       "      <td>10.681679</td>\n",
       "      <td>11.958755</td>\n",
       "      <td>9.379041</td>\n",
       "      <td>11.521346</td>\n",
       "      <td>9.096294</td>\n",
       "      <td>6.651197</td>\n",
       "      <td>9.080625</td>\n",
       "      <td>...</td>\n",
       "      <td>17.815496</td>\n",
       "      <td>18.115089</td>\n",
       "      <td>19.703150</td>\n",
       "      <td>17.968981</td>\n",
       "      <td>18.849733</td>\n",
       "      <td>18.742239</td>\n",
       "      <td>17.371330</td>\n",
       "      <td>18.414146</td>\n",
       "      <td>3</td>\n",
       "      <td>248.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Vascular smooth muscle contraction</th>\n",
       "      <td>7.945333</td>\n",
       "      <td>8.794459</td>\n",
       "      <td>8.044936</td>\n",
       "      <td>8.845881</td>\n",
       "      <td>7.499012</td>\n",
       "      <td>9.197755</td>\n",
       "      <td>10.165154</td>\n",
       "      <td>6.610901</td>\n",
       "      <td>5.228885</td>\n",
       "      <td>8.893410</td>\n",
       "      <td>...</td>\n",
       "      <td>14.953139</td>\n",
       "      <td>17.378485</td>\n",
       "      <td>18.899155</td>\n",
       "      <td>16.883688</td>\n",
       "      <td>17.594791</td>\n",
       "      <td>19.304136</td>\n",
       "      <td>16.785450</td>\n",
       "      <td>18.402016</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Nicotine addiction</th>\n",
       "      <td>7.692084</td>\n",
       "      <td>10.337347</td>\n",
       "      <td>5.269646</td>\n",
       "      <td>7.934579</td>\n",
       "      <td>8.343276</td>\n",
       "      <td>8.747339</td>\n",
       "      <td>9.423052</td>\n",
       "      <td>9.130812</td>\n",
       "      <td>7.127965</td>\n",
       "      <td>9.038717</td>\n",
       "      <td>...</td>\n",
       "      <td>10.675890</td>\n",
       "      <td>19.353052</td>\n",
       "      <td>18.349897</td>\n",
       "      <td>17.122292</td>\n",
       "      <td>18.314499</td>\n",
       "      <td>18.945776</td>\n",
       "      <td>15.731035</td>\n",
       "      <td>19.770882</td>\n",
       "      <td>2</td>\n",
       "      <td>97.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>320 rows × 15963 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    0          1          2  \\\n",
       "gene_set_name                                                                 \n",
       "Ras signaling pathway                       13.004891  18.604397  10.770545   \n",
       "Synthesis and degradation of ketone bodies  10.549258  11.033575   9.806051   \n",
       "Tryptophan metabolism                        8.511940  10.520400   8.525270   \n",
       "Cellular senescence                         10.276177  11.345402   9.506522   \n",
       "Melanoma                                    10.338791  11.616563   7.900808   \n",
       "...                                               ...        ...        ...   \n",
       "Fatty acid elongation                        8.012397   9.935248   5.185310   \n",
       "Mineral absorption                           7.773013  10.744399   6.619865   \n",
       "Legionellosis                               11.014463  11.877916  10.346815   \n",
       "Vascular smooth muscle contraction           7.945333   8.794459   8.044936   \n",
       "Nicotine addiction                           7.692084  10.337347   5.269646   \n",
       "\n",
       "                                                    3          4          5  \\\n",
       "gene_set_name                                                                 \n",
       "Ras signaling pathway                       11.742773  12.278329  13.361037   \n",
       "Synthesis and degradation of ketone bodies  10.190495   8.996559  10.496225   \n",
       "Tryptophan metabolism                       10.302014   7.914603  11.500504   \n",
       "Cellular senescence                         10.369909   9.442755  11.060408   \n",
       "Melanoma                                     9.676971  11.094644   9.397411   \n",
       "...                                               ...        ...        ...   \n",
       "Fatty acid elongation                        8.025328   7.645795   8.087513   \n",
       "Mineral absorption                           8.377044  10.248653   8.882352   \n",
       "Legionellosis                               10.681679  11.958755   9.379041   \n",
       "Vascular smooth muscle contraction           8.845881   7.499012   9.197755   \n",
       "Nicotine addiction                           7.934579   8.343276   8.747339   \n",
       "\n",
       "                                                    6          7         8  \\\n",
       "gene_set_name                                                                \n",
       "Ras signaling pathway                       14.020433  12.585869  9.968098   \n",
       "Synthesis and degradation of ketone bodies  10.523148   7.660634  5.964342   \n",
       "Tryptophan metabolism                       10.822841   8.972867  5.907670   \n",
       "Cellular senescence                         12.644639   7.495144  5.618484   \n",
       "Melanoma                                    11.215753   7.710507  4.410573   \n",
       "...                                               ...        ...       ...   \n",
       "Fatty acid elongation                        9.565404   7.208513  4.885524   \n",
       "Mineral absorption                           9.186312   6.799828  6.235319   \n",
       "Legionellosis                               11.521346   9.096294  6.651197   \n",
       "Vascular smooth muscle contraction          10.165154   6.610901  5.228885   \n",
       "Nicotine addiction                           9.423052   9.130812  7.127965   \n",
       "\n",
       "                                                    9  ...      15953  \\\n",
       "gene_set_name                                          ...              \n",
       "Ras signaling pathway                       14.672712  ...  18.874718   \n",
       "Synthesis and degradation of ketone bodies  10.937921  ...  13.176636   \n",
       "Tryptophan metabolism                       10.473205  ...  15.393194   \n",
       "Cellular senescence                         11.946845  ...  15.561532   \n",
       "Melanoma                                     9.654911  ...  18.076948   \n",
       "...                                               ...  ...        ...   \n",
       "Fatty acid elongation                        8.757915  ...  12.517259   \n",
       "Mineral absorption                           8.192952  ...  13.347853   \n",
       "Legionellosis                                9.080625  ...  17.815496   \n",
       "Vascular smooth muscle contraction           8.893410  ...  14.953139   \n",
       "Nicotine addiction                           9.038717  ...  10.675890   \n",
       "\n",
       "                                                15954      15955      15956  \\\n",
       "gene_set_name                                                                 \n",
       "Ras signaling pathway                       17.437302  11.635219  14.051967   \n",
       "Synthesis and degradation of ketone bodies  17.839787  18.019396  18.009800   \n",
       "Tryptophan metabolism                       16.794025  18.552992  19.267550   \n",
       "Cellular senescence                         18.979267  15.641437  17.741268   \n",
       "Melanoma                                    17.905724  19.393864  17.818716   \n",
       "...                                               ...        ...        ...   \n",
       "Fatty acid elongation                       17.370972  18.435896  16.272781   \n",
       "Mineral absorption                          16.770800  19.046362  18.393661   \n",
       "Legionellosis                               18.115089  19.703150  17.968981   \n",
       "Vascular smooth muscle contraction          17.378485  18.899155  16.883688   \n",
       "Nicotine addiction                          19.353052  18.349897  17.122292   \n",
       "\n",
       "                                                15957      15958      15959  \\\n",
       "gene_set_name                                                                 \n",
       "Ras signaling pathway                       10.695468  18.918125  15.753615   \n",
       "Synthesis and degradation of ketone bodies  16.384592  19.501698  13.787357   \n",
       "Tryptophan metabolism                       17.295149  18.762913  17.576466   \n",
       "Cellular senescence                         15.099967  20.585846  16.619169   \n",
       "Melanoma                                    17.743242  19.693615  16.468264   \n",
       "...                                               ...        ...        ...   \n",
       "Fatty acid elongation                       16.137571  18.849882  15.194713   \n",
       "Mineral absorption                          17.909298  18.988417  17.036463   \n",
       "Legionellosis                               18.849733  18.742239  17.371330   \n",
       "Vascular smooth muscle contraction          17.594791  19.304136  16.785450   \n",
       "Nicotine addiction                          18.314499  18.945776  15.731035   \n",
       "\n",
       "                                                15960  distance_from_geneset  \\\n",
       "gene_set_name                                                                  \n",
       "Ras signaling pathway                       18.735266                      2   \n",
       "Synthesis and degradation of ketone bodies  19.948223                      2   \n",
       "Tryptophan metabolism                       18.849489                      2   \n",
       "Cellular senescence                         20.295670                      2   \n",
       "Melanoma                                    20.080921                      2   \n",
       "...                                               ...                    ...   \n",
       "Fatty acid elongation                       19.197926                      2   \n",
       "Mineral absorption                          19.364380                      2   \n",
       "Legionellosis                               18.414146                      3   \n",
       "Vascular smooth muscle contraction          18.402016                      2   \n",
       "Nicotine addiction                          19.770882                      2   \n",
       "\n",
       "                                             rank  \n",
       "gene_set_name                                      \n",
       "Ras signaling pathway                        97.0  \n",
       "Synthesis and degradation of ketone bodies   97.0  \n",
       "Tryptophan metabolism                        97.0  \n",
       "Cellular senescence                          97.0  \n",
       "Melanoma                                     97.0  \n",
       "...                                           ...  \n",
       "Fatty acid elongation                        97.0  \n",
       "Mineral absorption                           97.0  \n",
       "Legionellosis                               248.5  \n",
       "Vascular smooth muscle contraction           97.0  \n",
       "Nicotine addiction                           97.0  \n",
       "\n",
       "[320 rows x 15963 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "p2g_df['distance_from_geneset'] = distance_from_geneset_df\n",
    "p2g_df['rank']=p2g_df['distance_from_geneset'].rank()\n",
    "nearest_list = p2g_df.nsmallest(20,'rank').index\n",
    "p2g_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "68149521",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-05-22T17:44:43.930955Z",
     "start_time": "2022-05-22T17:44:43.915995Z"
    },
    "hidden": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Ras signaling pathway\n",
      "Synthesis and degradation of ketone bodies\n",
      "Tryptophan metabolism\n",
      "Cellular senescence\n",
      "Melanoma\n",
      "Pentose and glucuronate interconversions\n",
      "Chemical carcinogenesis\n",
      "RNA polymerase\n",
      "Antigen processing and presentation\n",
      "N-Glycan biosynthesis\n",
      "Type I diabetes mellitus\n",
      "Long-term depression\n",
      "Long-term potentiation\n",
      "Mitophagy\n",
      "Kaposi sarcoma-associated herpesvirus infection\n",
      "Spinocerebellar ataxia\n",
      "Propanoate metabolism\n",
      "Hypertrophic cardiomyopathy\n",
      "Proteoglycans in cancer\n",
      "Th1 and Th2 cell differentiation\n"
     ]
    }
   ],
   "source": [
    "for i in nearest_list:\n",
    "    print(i)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f60f91d5",
   "metadata": {},
   "source": [
    "# function "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "42f4a250",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:23:13.295408Z",
     "start_time": "2022-08-11T11:23:13.280450Z"
    }
   },
   "outputs": [],
   "source": [
    "mu,p2g_df,g2g_dataframe\n",
    "\n",
    "def pathway_prioritization(gene_set,method,number):\n",
    "    g2g_dataframe_disease_pathway = g2g_dataframe[g2g_dataframe.index.isin(gene_set)]\n",
    "    \n",
    "    def gene_set_mean(g2g_dataframe_disease_pathway):\n",
    "        df = copy.deepcopy(g2g_dataframe_disease_pathway)\n",
    "        value = df.mean()\n",
    "        return value\n",
    "    def gene_set_median(g2g_dataframe_disease_pathway):\n",
    "        df = copy.deepcopy(g2g_dataframe_disease_pathway)\n",
    "        value = df.median()\n",
    "        return value\n",
    "    if method == 'mean':\n",
    "        value = gene_set_mean(g2g_dataframe_disease_pathway)\n",
    "    elif method=='median':\n",
    "        value = gene_set_median(g2g_dataframe_disease_pathway)\n",
    "    #median = gene_set_median(g2g_dataframe_disease_pathway)\n",
    "    DiseaseAndPathway_mu_distance = []\n",
    "    for i in tqdm(range(len(mu))):\n",
    "        #mean = gene_set_mean(g2g_dataframe_disease_pathway)\n",
    "        norm = np.linalg.norm(value - mu[i])\n",
    "        DiseaseAndPathway_mu_distance.append(norm)\n",
    "        \n",
    "    distance_from_geneset_df = copy.deepcopy(DiseaseAndPathway_mu_distance)\n",
    "    distance_from_geneset_df = list(map(int,distance_from_geneset_df))\n",
    "    p2g_df['distance_from_geneset'] = distance_from_geneset_df\n",
    "    p2g_df['rank']=p2g_df['distance_from_geneset'].rank()\n",
    "    nearest_list = list(p2g_df.nsmallest(number,'rank').index)\n",
    "    return nearest_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "1a43da16",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:23:14.134581Z",
     "start_time": "2022-08-11T11:23:14.032322Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 8443.97it/s]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "['Ras signaling pathway',\n",
       " 'Tryptophan metabolism',\n",
       " 'Cellular senescence',\n",
       " 'Melanoma',\n",
       " 'Antigen processing and presentation',\n",
       " 'Long-term depression',\n",
       " 'Mitophagy',\n",
       " 'Propanoate metabolism',\n",
       " 'Th1 and Th2 cell differentiation',\n",
       " 'Parkinson disease',\n",
       " 'Thyroid cancer',\n",
       " 'Intestinal immune network for IgA production',\n",
       " 'IL-17 signaling pathway',\n",
       " 'Hippo signaling pathway',\n",
       " 'Prolactin signaling pathway',\n",
       " 'Sphingolipid signaling pathway',\n",
       " 'Graft-versus-host disease',\n",
       " 'GnRH secretion',\n",
       " 'Alanine, aspartate and glutamate metabolism',\n",
       " 'Synaptic vesicle cycle']"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathway_prioritization(gene_set_from_cancer_type[0][0],mu,p2g_df,g2g_dataframe)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "8a14fc12",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:24:20.728721Z",
     "start_time": "2022-08-11T11:24:19.967756Z"
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 8912.61it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 8679.37it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 8912.67it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 9167.06it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 8912.79it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 9167.31it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 7825.83it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 8671.91it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 8912.73it/s]\n",
      "100%|██████████████████████████████████████████████████████████████████████████████| 320/320 [00:00<00:00, 8912.61it/s]\n"
     ]
    }
   ],
   "source": [
    "acc_pathway = []\n",
    "for i in range(len(gene_set_from_cancer_type[0])):\n",
    "    acc_pathway.append(pathway_prioritization(gene_set_from_cancer_type[0][i],mu,p2g_df,g2g_dataframe))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "f265ff2e",
   "metadata": {
    "ExecuteTime": {
     "end_time": "2022-08-11T11:42:59.838563Z",
     "start_time": "2022-08-11T11:42:59.829588Z"
    }
   },
   "outputs": [],
   "source": [
    "with open('D://downloads/acc_enrich_result_by_graphembedding.csv','w') as acc_output:\n",
    "    for j in range(len(number_from_cancer_type_count[0])):\n",
    "        acc_output.write(driverdb.tool[j])\n",
    "        acc_output.write(',')\n",
    "        for i in range(20):\n",
    "            acc_output.write(acc_pathway[j][i])\n",
    "            acc_output.write(',')\n",
    "        acc_output.write('\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d08517f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
